diff --git a/cmake/modules/HandleLLVMOptions.cmake b/cmake/modules/HandleLLVMOptions.cmake
index 7b8edecd..d2c860ac 100644
--- a/cmake/modules/HandleLLVMOptions.cmake
+++ b/cmake/modules/HandleLLVMOptions.cmake
@@ -223,6 +223,10 @@ endif( CMAKE_SIZEOF_VOID_P EQUAL 8 AND NOT WIN32 )
 
 if (LLVM_BUILD_STATIC)
   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -static")
+  if(MINGW)
+    # We want to mimic MSVC build here
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -static")
+  endif()
 endif()
 
 if( XCODE )
diff --git a/tools/clang/CMakeLists.txt b/tools/clang/CMakeLists.txt
index 71190336..328ffad8 100644
--- a/tools/clang/CMakeLists.txt
+++ b/tools/clang/CMakeLists.txt
@@ -431,6 +431,9 @@ install(DIRECTORY include/clang-c
   )
 
 add_definitions( -D_GNU_SOURCE )
+if (MINGW)
+  add_definitions( -DMS_SUPPORT_VARIABLE_LANGOPTS )
+endif()
 
 option(CLANG_ENABLE_ARCMT "Build ARCMT." ON)
 if (CLANG_ENABLE_ARCMT)
