diff --git a/include/dxc/Support/Global.h b/include/dxc/Support/Global.h
index 9d97467d..2262b464 100644
--- a/include/dxc/Support/Global.h
+++ b/include/dxc/Support/Global.h
@@ -281,7 +281,7 @@ inline void OutputDebugFormatA(const char *pszFormat, ...) {
 
 #ifndef NDEBUG
 
-#ifdef _WIN32
+#ifdef _MSC_VER
 
 // DXASSERT is used to debug break when 'exp' evaluates to false and is only
 //     intended for internal developer use. It is compiled out in free
@@ -314,7 +314,7 @@ inline void OutputDebugFormatA(const char *pszFormat, ...) {
 
 #define DXVERIFY_NOMSG(exp) DXASSERT(exp, "")
 
-#else // _WIN32
+#else // _MSC_VER
 #include <cassert>
 
 #define DXASSERT_NOMSG assert
@@ -341,7 +341,7 @@ inline void OutputDebugFormatA(const char *pszFormat, ...) {
     }                                                                          \
   } while (0)
 
-#endif // _WIN32
+#endif // _MSC_VER
 
 #else // NDEBUG
 
diff --git a/include/dxc/Support/WinIncludes.h b/include/dxc/Support/WinIncludes.h
index b1c25ec3..854e3615 100644
--- a/include/dxc/Support/WinIncludes.h
+++ b/include/dxc/Support/WinIncludes.h
@@ -65,6 +65,7 @@ template <class T> void swap(CComHeapPtr<T> &a, CComHeapPtr<T> &b) {
 
 #include "dxc/WinAdapter.h"
 
+#ifndef __MINGW32__
 #ifdef __cplusplus
 #if !defined(DEFINE_ENUM_FLAG_OPERATORS)
 // Define operator overloads to enable bit operations on enum values that are
@@ -118,6 +119,7 @@ template <class T> struct _ENUM_FLAG_SIZED_INTEGER {
 #else
 #define DEFINE_ENUM_FLAG_OPERATORS(ENUMTYPE) // NOP, C allows these operators.
 #endif
+#endif // __MINGW32__
 
 #endif // _MSC_VER
 
diff --git a/include/dxc/WinAdapter.h b/include/dxc/WinAdapter.h
index b8c66468..70b7cad6 100644
--- a/include/dxc/WinAdapter.h
+++ b/include/dxc/WinAdapter.h
@@ -15,7 +15,7 @@
 #ifndef LLVM_SUPPORT_WIN_ADAPTER_H
 #define LLVM_SUPPORT_WIN_ADAPTER_H
 
-#ifndef _WIN32
+#ifndef _MSC_VER
 
 #ifdef __cplusplus
 #include <atomic>
@@ -33,6 +33,137 @@
 #include <vector>
 #endif // __cplusplus
 
+#ifdef __MINGW32__
+
+#undef _WIN32_WINNT
+#undef _WIN32_IE
+
+// Require at least Windows 7 (Updated from XP)
+#define _WIN32_WINNT 0x0601
+#define _WIN32_IE 0x0800
+
+#define WIN32_LEAN_AND_MEAN
+#define STRSAFE_NO_DEPRECATE
+
+#include <intsafe.h>
+#include <objidl.h>
+#include <sal.h>
+#include <strsafe.h>
+#include <unknwn.h>
+#include <windows.h>
+
+#undef EN
+#undef IN
+#undef OUT
+#undef MemoryFence
+#undef ReplaceText
+
+#undef UNREFERENCED_PARAMETER
+#define UNREFERENCED_PARAMETER(P) (void)(P)
+
+#ifndef ERROR_NOT_CAPABLE
+#define ERROR_NOT_CAPABLE __MSABI_LONG(775)
+#endif
+#ifndef ERROR_UNHANDLED_EXCEPTION
+#define ERROR_UNHANDLED_EXCEPTION __MSABI_LONG(574)
+#endif
+
+#define EventRegisterMicrosoft_Windows_DXCompiler_API()
+#define EventUnregisterMicrosoft_Windows_DXCompiler_API()
+
+#define DxcEtw_DXCompilerCompile_Start()
+#define DxcEtw_DXCompilerCompile_Stop(hr)
+#define DxcEtw_DXCompilerCreateInstance_Start()
+#define DxcEtw_DXCompilerCreateInstance_Stop(hr)
+#define DxcEtw_DXCompilerDisassemble_Start()
+#define DxcEtw_DXCompilerDisassemble_Stop(hr)
+#define DxcEtw_DXCompilerInitialization_Start()
+#define DxcEtw_DXCompilerInitialization_Stop(hr)
+#define DxcEtw_DXCompilerPreprocess_Start()
+#define DxcEtw_DXCompilerPreprocess_Stop(hr)
+#define DxcEtw_DXCompilerShutdown_Start()
+#define DxcEtw_DXCompilerShutdown_Stop(hr)
+#define DxcEtw_DxcValidation_Start()
+#define DxcEtw_DxcValidation_Stop(hr)
+
+#define ATLASSERT assert
+
+// Disable calling convention attributes
+#define __cdecl
+
+// FIXME: missing from intsafe.h
+#define Int32ToUInt32 IntToUInt
+#define UInt32Add UIntAdd
+#define UInt32Mult UIntMult
+
+// FIXME: missing from winerror.h
+#define E_BOUNDS _HRESULT_TYPEDEF_(0x8000000B)
+#define E_NOT_VALID_STATE HRESULT_FROM_WIN32(ERROR_INVALID_STATE)
+
+// FIXME: missing from sal.h
+#define _Maybenull_
+#define _Notnull_
+
+// FIXME: missing from d3d12shader.h
+#define D3D_SHADER_REQUIRES_EARLY_DEPTH_STENCIL 0x00000002
+#define D3D12_SHVER_GET_TYPE(_Version) (((_Version) >> 16) & 0xffff)
+#define D3D12_SHVER_GET_MAJOR(_Version) (((_Version) >> 4) & 0xf)
+#define D3D12_SHVER_GET_MINOR(_Version) (((_Version) >> 0) & 0xf)
+
+#ifdef __cplusplus
+
+constexpr uint8_t nybble_from_hex(char c) {
+  return ((c >= '0' && c <= '9')
+              ? (c - '0')
+              : ((c >= 'a' && c <= 'f')
+                     ? (c - 'a' + 10)
+                     : ((c >= 'A' && c <= 'F') ? (c - 'A' + 10)
+                                               : /* Should be an error */ -1)));
+}
+
+constexpr uint8_t byte_from_hex(char c1, char c2) {
+  return nybble_from_hex(c1) << 4 | nybble_from_hex(c2);
+}
+
+constexpr uint8_t byte_from_hexstr(const char str[2]) {
+  return nybble_from_hex(str[0]) << 4 | nybble_from_hex(str[1]);
+}
+
+constexpr GUID guid_from_string(const char str[37]) {
+  return GUID{static_cast<uint32_t>(byte_from_hexstr(str)) << 24 |
+                  static_cast<uint32_t>(byte_from_hexstr(str + 2)) << 16 |
+                  static_cast<uint32_t>(byte_from_hexstr(str + 4)) << 8 |
+                  byte_from_hexstr(str + 6),
+              static_cast<uint16_t>(
+                  static_cast<uint16_t>(byte_from_hexstr(str + 9)) << 8 |
+                  byte_from_hexstr(str + 11)),
+              static_cast<uint16_t>(
+                  static_cast<uint16_t>(byte_from_hexstr(str + 14)) << 8 |
+                  byte_from_hexstr(str + 16)),
+              {byte_from_hexstr(str + 19), byte_from_hexstr(str + 21),
+               byte_from_hexstr(str + 24), byte_from_hexstr(str + 26),
+               byte_from_hexstr(str + 28), byte_from_hexstr(str + 30),
+               byte_from_hexstr(str + 32), byte_from_hexstr(str + 34)}};
+}
+
+#define CROSS_PLATFORM_UUIDOF(name, spec)                                      \
+  struct __declspec(uuid(spec)) name;                                          \
+  extern "C++" {                                                               \
+  template <> struct __mingw_uuidof_s<name> {                                  \
+    static constexpr IID __uuid_inst = guid_from_string(spec);                 \
+  };                                                                           \
+  template <> constexpr const GUID &__mingw_uuidof<name>() {                   \
+    return __mingw_uuidof_s<name>::__uuid_inst;                                \
+  }                                                                            \
+  template <> constexpr const GUID &__mingw_uuidof<name *>() {                 \
+    return __mingw_uuidof_s<name>::__uuid_inst;                                \
+  }                                                                            \
+  }
+
+#endif // __cplusplus
+
+#else // __MINGW32__
+
 #define COM_NO_WINDOWS_H // needed to inform d3d headers that this isn't windows
 
 //===----------------------------------------------------------------------===//
@@ -630,6 +761,12 @@ CROSS_PLATFORM_UUIDOF(ID3D12LibraryReflection,
 CROSS_PLATFORM_UUIDOF(ID3D12ShaderReflection,
                       "5A58797D-A72C-478D-8BA2-EFC6B0EFE88E")
 
+#endif // __cplusplus
+
+#endif // __MINGW32__
+
+#ifdef __cplusplus
+
 //===--------------------- COM Pointer Types ------------------------------===//
 
 class CAllocator {
@@ -897,6 +1034,7 @@ public:
 
 #define CComHeapPtr CHeapPtr
 
+#ifndef __MINGW32__
 //===--------------------------- BSTR Allocation --------------------------===//
 
 void SysFreeString(BSTR bstrString);
@@ -905,6 +1043,7 @@ BSTR SysAllocStringLen(const OLECHAR *strIn, UINT ui);
 
 //===--------------------------- BSTR Length ------------------------------===//
 unsigned int SysStringLen(const BSTR bstrString);
+#endif // __MINGW32__
 
 //===--------------------- UTF-8 Related Types ----------------------------===//
 
@@ -1034,6 +1173,6 @@ public:
 
 #endif // __cplusplus
 
-#endif // _WIN32
+#endif // _MSC_VER
 
 #endif // LLVM_SUPPORT_WIN_ADAPTER_H
diff --git a/include/dxc/dxcapi.h b/include/dxc/dxcapi.h
index 752041b7..330ba873 100644
--- a/include/dxc/dxcapi.h
+++ b/include/dxc/dxcapi.h
@@ -13,7 +13,7 @@
 #ifndef __DXC_API__
 #define __DXC_API__
 
-#ifdef _WIN32
+#ifdef _MSC_VER
 #ifndef DXC_API_IMPORT
 #define DXC_API_IMPORT __declspec(dllimport)
 #endif
@@ -1222,7 +1222,7 @@ struct IDxcPdbUtils2 : public IUnknown {
 // Note: __declspec(selectany) requires 'extern'
 // On Linux __declspec(selectany) is removed and using 'extern' results in link
 // error.
-#ifdef _MSC_VER
+#ifdef _WIN32
 #define CLSID_SCOPE __declspec(selectany) extern
 #else
 #define CLSID_SCOPE
diff --git a/include/dxc/dxcdxrfallbackcompiler.h b/include/dxc/dxcdxrfallbackcompiler.h
index c15906ca..d41872ac 100644
--- a/include/dxc/dxcdxrfallbackcompiler.h
+++ b/include/dxc/dxcdxrfallbackcompiler.h
@@ -12,7 +12,8 @@
 
 #ifndef __DXC_DXR_FALLBACK_COMPILER_API__
 #define __DXC_DXR_FALLBACK_COMPILER_API__
-#include "dxcapi.h"
+
+#include "dxc/dxcapi.h"
 
 enum class ShaderType : unsigned int {
   Raygen,
@@ -94,7 +95,7 @@ struct __declspec(uuid("76bb3c85-006d-4b72-9e10-63cd97df57f0"))
 // Note: __declspec(selectany) requires 'extern'
 // On Linux __declspec(selectany) is removed and using 'extern' results in link
 // error.
-#ifdef _MSC_VER
+#ifdef _WIN32
 #define CLSID_SCOPE __declspec(selectany) extern
 #else
 #define CLSID_SCOPE
diff --git a/include/dxc/dxcisense.h b/include/dxc/dxcisense.h
index 661de168..6b6dbd33 100644
--- a/include/dxc/dxcisense.h
+++ b/include/dxc/dxcisense.h
@@ -12,7 +12,7 @@
 #ifndef __DXC_ISENSE__
 #define __DXC_ISENSE__
 
-#include "dxcapi.h"
+#include "dxc/dxcapi.h"
 #ifndef _WIN32
 #include "WinAdapter.h"
 #endif
@@ -943,7 +943,7 @@ struct IDxcCompletionString : public IUnknown {
 // CLSID_DxcIntelliSense is not visible externally (this is OK in C, since const
 // is not by default static in C)
 
-#ifdef _MSC_VER
+#ifdef _WIN32
 #define CLSID_SCOPE __declspec(selectany) extern
 #else
 #define CLSID_SCOPE
diff --git a/include/dxc/dxctools.h b/include/dxc/dxctools.h
index bfdc0d86..bacc20c1 100644
--- a/include/dxc/dxctools.h
+++ b/include/dxc/dxctools.h
@@ -42,7 +42,7 @@ struct IDxcRewriter : public IUnknown {
       IDxcOperationResult **ppResult) = 0;
 };
 
-#ifdef _MSC_VER
+#ifdef _WIN32
 #define CLSID_SCOPE __declspec(selectany) extern
 #else
 #define CLSID_SCOPE
diff --git a/include/llvm/ADT/StringRef.h b/include/llvm/ADT/StringRef.h
index 957707f4..f5a49a4c 100644
--- a/include/llvm/ADT/StringRef.h
+++ b/include/llvm/ADT/StringRef.h
@@ -12,6 +12,7 @@
 
 #include <algorithm>
 #include <cassert>
+#include <cstdint>
 #include <cstring>
 #include <limits>
 #include <string>
