diff --git a/include/llvm/Object/COFF.h b/include/llvm/Object/COFF.h
index 025a9dbc..723c8b77 100644
--- a/include/llvm/Object/COFF.h
+++ b/include/llvm/Object/COFF.h
@@ -326,7 +326,7 @@ public:
   uint8_t getBaseType() const { return getType() & 0x0F; }
 
   uint8_t getComplexType() const {
-    return (getType() & 0xF0) >> COFF::SCT_COMPLEX_TYPE_SHIFT;
+    return (getType() & 0xF0) >> ::COFF::SCT_COMPLEX_TYPE_SHIFT;
   }
 
   bool isAbsolute() const {
@@ -334,31 +334,31 @@ public:
   }
 
   bool isExternal() const {
-    return getStorageClass() == COFF::IMAGE_SYM_CLASS_EXTERNAL;
+    return getStorageClass() == ::COFF::IMAGE_SYM_CLASS_EXTERNAL;
   }
 
   bool isCommon() const {
-    return isExternal() && getSectionNumber() == COFF::IMAGE_SYM_UNDEFINED &&
+    return isExternal() && getSectionNumber() == ::COFF::IMAGE_SYM_UNDEFINED &&
            getValue() != 0;
   }
 
   bool isUndefined() const {
-    return isExternal() && getSectionNumber() == COFF::IMAGE_SYM_UNDEFINED &&
+    return isExternal() && getSectionNumber() == ::COFF::IMAGE_SYM_UNDEFINED &&
            getValue() == 0;
   }
 
   bool isWeakExternal() const {
-    return getStorageClass() == COFF::IMAGE_SYM_CLASS_WEAK_EXTERNAL;
+    return getStorageClass() == ::COFF::IMAGE_SYM_CLASS_WEAK_EXTERNAL;
   }
 
   bool isFunctionDefinition() const {
-    return isExternal() && getBaseType() == COFF::IMAGE_SYM_TYPE_NULL &&
-           getComplexType() == COFF::IMAGE_SYM_DTYPE_FUNCTION &&
+    return isExternal() && getBaseType() == ::COFF::IMAGE_SYM_TYPE_NULL &&
+           getComplexType() == ::COFF::IMAGE_SYM_DTYPE_FUNCTION &&
            !COFF::isReservedSectionNumber(getSectionNumber());
   }
 
   bool isFunctionLineInfo() const {
-    return getStorageClass() == COFF::IMAGE_SYM_CLASS_FUNCTION;
+    return getStorageClass() == ::COFF::IMAGE_SYM_CLASS_FUNCTION;
   }
 
   bool isAnyUndefined() const {
@@ -366,27 +366,27 @@ public:
   }
 
   bool isFileRecord() const {
-    return getStorageClass() == COFF::IMAGE_SYM_CLASS_FILE;
+    return getStorageClass() == ::COFF::IMAGE_SYM_CLASS_FILE;
   }
 
   bool isSection() const {
-    return getStorageClass() == COFF::IMAGE_SYM_CLASS_SECTION;
+    return getStorageClass() == ::COFF::IMAGE_SYM_CLASS_SECTION;
   }
 
   bool isSectionDefinition() const {
     // C++/CLI creates external ABS symbols for non-const appdomain globals.
     // These are also followed by an auxiliary section definition.
     bool isAppdomainGlobal =
-        getStorageClass() == COFF::IMAGE_SYM_CLASS_EXTERNAL &&
-        getSectionNumber() == COFF::IMAGE_SYM_ABSOLUTE;
-    bool isOrdinarySection = getStorageClass() == COFF::IMAGE_SYM_CLASS_STATIC;
+        getStorageClass() == ::COFF::IMAGE_SYM_CLASS_EXTERNAL &&
+        getSectionNumber() == ::COFF::IMAGE_SYM_ABSOLUTE;
+    bool isOrdinarySection = getStorageClass() == ::COFF::IMAGE_SYM_CLASS_STATIC;
     if (!getNumberOfAuxSymbols())
       return false;
     return isAppdomainGlobal || isOrdinarySection;
   }
 
   bool isCLRToken() const {
-    return getStorageClass() == COFF::IMAGE_SYM_CLASS_CLR_TOKEN;
+    return getStorageClass() == ::COFF::IMAGE_SYM_CLASS_CLR_TOKEN;
   }
 
 private:
@@ -411,7 +411,7 @@ struct coff_section {
   // Returns true if the actual number of relocations is stored in
   // VirtualAddress field of the first relocation table entry.
   bool hasExtendedRelocations() const {
-    return (Characteristics & COFF::IMAGE_SCN_LNK_NRELOC_OVFL) &&
+    return (Characteristics & ::COFF::IMAGE_SCN_LNK_NRELOC_OVFL) &&
            NumberOfRelocations == UINT16_MAX;
   }
 };
diff --git a/include/llvm/Support/COFF.h b/include/llvm/Support/COFF.h
index 3c5ee069..0a81bd07 100644
--- a/include/llvm/Support/COFF.h
+++ b/include/llvm/Support/COFF.h
@@ -27,6 +27,11 @@
 #include <cassert>
 #include <cstring>
 
+// Prevent inclusion of winnt.h
+#ifndef _WINNT_
+#define _WINNT_
+#endif
+
 namespace llvm {
 namespace COFF {
 
@@ -83,9 +88,9 @@ namespace COFF {
   enum MachineTypes {
     MT_Invalid = 0xffff,
 
-    IMAGE_FILE_MACHINE_UNKNOWN   = 0x0,
-    IMAGE_FILE_MACHINE_AM33      = 0x13,
-    IMAGE_FILE_MACHINE_AMD64     = 0x8664,
+    COFF::IMAGE_FILE_MACHINE_UNKNOWN   = 0x0,
+    ::IMAGE_FILE_MACHINE_AM33      = 0x13,
+    ::IMAGE_FILE_MACHINE_AMD64     = 0x8664,
     IMAGE_FILE_MACHINE_ARM       = 0x1C0,
     IMAGE_FILE_MACHINE_ARMNT     = 0x1C4,
     IMAGE_FILE_MACHINE_EBC       = 0xEBC,
