diff --git a/src/3rdparty/chromium/base/BUILD.gn b/src/3rdparty/chromium/base/BUILD.gn
index 6b87aad..d5091a4 100644
--- a/src/3rdparty/chromium/base/BUILD.gn
+++ b/src/3rdparty/chromium/base/BUILD.gn
@@ -144,6 +144,7 @@ if (is_win) {
   # This is in a separate config so the flags can be applied to dependents.
   # ldflags in GN aren't automatically inherited.
   config("base_win_linker_flags") {
+    if (!is_mingw) {
     ldflags = [
       # Linking with shcore.lib causes the shcore api forwarder dll to load.
       "/DELAYLOAD:api-ms-win-shcore-scaling-l1-1-1.dll",
@@ -157,6 +158,7 @@ if (is_win) {
       "/DELAYLOAD:setupapi.dll",
       "/DELAYLOAD:shcore.dll",
     ]
+    }
   }
 }
 
@@ -1911,6 +1913,22 @@ jumbo_component("base") {
       "win/wrapped_window_proc.h",
     ]
 
+    if (is_mingw) {
+      sources -= [
+        "win/atl.h",                # requires <atlbase.h>
+        "win/atl_throw.cc",         # requires <atlbase.h>
+        "win/atl_throw.h",
+        "win/com_init_balancer.cc", # requires <wrl/implements.h>
+        "win/com_init_balancer.h",
+        "win/enum_variant.cc",      # requires <wrl/implements.h>
+        "win/enum_variant.h",
+        "win/vector.cc",            # requires <wrl/implements.h>
+        "win/vector.h",
+      ]
+
+      defines += [ "STRSAFE_NO_DEPRECATE" ]
+    }
+
     # winternl.h and NTSecAPI.h have different definitions of UNICODE_STRING.
     # There's only one client of NTSecAPI.h in base but several of winternl.h,
     # so exclude the NTSecAPI.h one.
@@ -1923,16 +1941,16 @@ jumbo_component("base") {
     }
 
     libs += [
-      "cfgmgr32.lib",
-      "ntdll.lib",
-      "onecore.lib",
-      "powrprof.lib",
-      "propsys.lib",
-      "setupapi.lib",
-      "shcore.lib",
-      "userenv.lib",
-      "wbemuuid.lib",
-      "winmm.lib",
+      "cfgmgr32",
+      "ntdll",
+      "onecore",
+      "powrprof",
+      "propsys",
+      "setupapi",
+      "shcore",
+      "userenv",
+      "wbemuuid",
+      "winmm",
     ]
     all_dependent_configs += [
       ":base_win_linker_flags",
@@ -2396,7 +2414,7 @@ jumbo_component("base") {
       "//third_party/perfetto/gn:public_config",
     ]
 
-    if (is_win) {
+    if (is_win && !is_mingw) {
       sources += [
         "trace_event/trace_event_etw_export_win.cc",
         "trace_event/trace_event_etw_export_win.h",
@@ -2815,14 +2833,18 @@ if (is_win) {
   # base/test/data/pe_image.
   shared_library("pe_image_test") {
     sources = [ "win/pe_image_test.cc" ]
+    if (is_mingw) {
+      ldflags = [ "-mwindows" ]
+    } else {
     ldflags = [
       "/DELAYLOAD:cfgmgr32.dll",
       "/DELAYLOAD:shell32.dll",
       "/SUBSYSTEM:WINDOWS",
     ]
+    }
     libs = [
-      "cfgmgr32.lib",
-      "shell32.lib",
+      "cfgmgr32",
+      "shell32",
     ]
   }
 
diff --git a/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc_base/win/windows_types.h b/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc_base/win/windows_types.h
index 2171892..866c75f 100644
--- a/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc_base/win/windows_types.h
+++ b/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc_base/win/windows_types.h
@@ -19,8 +19,8 @@ extern "C" {
 
 typedef unsigned long DWORD;
 typedef long LONG;
-typedef __int64 LONGLONG;
-typedef unsigned __int64 ULONGLONG;
+typedef int64_t LONGLONG;
+typedef uint64_t ULONGLONG;
 
 #define VOID void
 typedef char CHAR;
@@ -29,7 +29,7 @@ typedef long LONG;
 typedef int INT;
 typedef unsigned int UINT;
 typedef unsigned int* PUINT;
-typedef unsigned __int64 UINT64;
+typedef uint64_t UINT64;
 typedef void* LPVOID;
 typedef void* PVOID;
 typedef void* HANDLE;
@@ -61,12 +61,18 @@ struct PA_CHROME_SRWLOCK {
 
 // Define some macros needed when prototyping Windows functions.
 
+#ifndef DECLSPEC_IMPORT
 #define DECLSPEC_IMPORT __declspec(dllimport)
+#endif
 #define WINBASEAPI DECLSPEC_IMPORT
 #define WINAPI __stdcall
 
 // Needed for LockImpl.
+#if defined(COMPILER_MSVC)
 WINBASEAPI _Releases_exclusive_lock_(*SRWLock) VOID WINAPI
+#else
+VOID WINAPI
+#endif
     ReleaseSRWLockExclusive(_Inout_ PSRWLOCK SRWLock);
 WINBASEAPI BOOLEAN WINAPI TryAcquireSRWLockExclusive(_Inout_ PSRWLOCK SRWLock);
 
diff --git a/src/3rdparty/chromium/base/debug/handle_hooks_win.cc b/src/3rdparty/chromium/base/debug/handle_hooks_win.cc
index 9bb8907..0741259 100644
--- a/src/3rdparty/chromium/base/debug/handle_hooks_win.cc
+++ b/src/3rdparty/chromium/base/debug/handle_hooks_win.cc
@@ -9,6 +9,8 @@
 #include <psapi.h>
 #include <stddef.h>
 
+#include <memory>
+
 #include "base/logging.h"
 #include "base/memory/raw_ptr.h"
 #include "base/numerics/safe_conversions.h"
diff --git a/src/3rdparty/chromium/base/enterprise_util_win.cc b/src/3rdparty/chromium/base/enterprise_util_win.cc
index 1c0965f..39d6ad4 100644
--- a/src/3rdparty/chromium/base/enterprise_util_win.cc
+++ b/src/3rdparty/chromium/base/enterprise_util_win.cc
@@ -16,14 +16,22 @@ bool IsManagedDevice() {
   // they are managed.
   // In addition, simply being joined to AAD does not mean the device is being
   // managed by the AAD tenant, so checking for AAD join is not included here.
+#if defined(COMPILER_MSVC)
   return base::win::IsEnrolledToDomain() ||
          base::win::IsDeviceRegisteredWithManagement();
+#else
+  return base::win::IsEnrolledToDomain();
+#endif
 }
 
 bool IsEnterpriseDevice() {
   // Both legacy domain join and AAD join represent machine-wide enterprise
   // join.
+#if defined(COMPILER_MSVC)
   return base::win::IsEnrolledToDomain() || base::win::IsJoinedToAzureAD();
+#else
+  return base::win::IsEnrolledToDomain();
+#endif
 }
 
 }  // namespace base
diff --git a/src/3rdparty/chromium/base/export_template.h b/src/3rdparty/chromium/base/export_template.h
index 248d0d9..d5eed60 100644
--- a/src/3rdparty/chromium/base/export_template.h
+++ b/src/3rdparty/chromium/base/export_template.h
@@ -138,7 +138,9 @@
 
 EXPORT_TEMPLATE_TEST(DEFAULT, );
 EXPORT_TEMPLATE_TEST(DEFAULT, __attribute__((visibility("default"))));
+#if defined(COMPILER_MSVC)
 EXPORT_TEMPLATE_TEST(EXPORT_DLLEXPORT, __declspec(dllexport));
+#endif
 EXPORT_TEMPLATE_TEST(DEFAULT, __declspec(dllimport));
 
 #undef EXPORT_TEMPLATE_TEST
diff --git a/src/3rdparty/chromium/base/lazy_instance.h b/src/3rdparty/chromium/base/lazy_instance.h
index 5081e97..36c31d7 100644
--- a/src/3rdparty/chromium/base/lazy_instance.h
+++ b/src/3rdparty/chromium/base/lazy_instance.h
@@ -174,7 +174,7 @@ class LazyInstance {
   // MSVC gives a warning that the alignment expands the size of the
   // LazyInstance struct to make the size a multiple of the alignment. This
   // is expected in this case.
-#if BUILDFLAG(IS_WIN)
+#if defined(COMPILER_MSVC)
 #pragma warning(push)
 #pragma warning(disable : 4324)
 #endif
@@ -187,7 +187,7 @@ class LazyInstance {
   // Preallocated space for the Type instance.
   alignas(Type) char private_buf_[sizeof(Type)];
 
-#if BUILDFLAG(IS_WIN)
+#if defined(COMPILER_MSVC)
 #pragma warning(pop)
 #endif
 
diff --git a/src/3rdparty/chromium/base/memory/aligned_memory.cc b/src/3rdparty/chromium/base/memory/aligned_memory.cc
index 40d0e40..5eafbb0 100644
--- a/src/3rdparty/chromium/base/memory/aligned_memory.cc
+++ b/src/3rdparty/chromium/base/memory/aligned_memory.cc
@@ -19,7 +19,7 @@ void* AlignedAlloc(size_t size, size_t alignment) {
   DCHECK(bits::IsPowerOfTwo(alignment));
   DCHECK_EQ(alignment % sizeof(void*), 0U);
   void* ptr = nullptr;
-#if defined(COMPILER_MSVC)
+#if BUILDFLAG(IS_WIN)
   ptr = _aligned_malloc(size, alignment);
 #elif BUILDFLAG(IS_ANDROID)
   // Android technically supports posix_memalign(), but does not expose it in
diff --git a/src/3rdparty/chromium/base/power_monitor/speed_limit_observer_win.cc b/src/3rdparty/chromium/base/power_monitor/speed_limit_observer_win.cc
index 2454140..29c7969 100644
--- a/src/3rdparty/chromium/base/power_monitor/speed_limit_observer_win.cc
+++ b/src/3rdparty/chromium/base/power_monitor/speed_limit_observer_win.cc
@@ -20,7 +20,9 @@
 namespace {
 
 // From ntdef.f
+#ifndef NT_SUCCESS
 #define NT_SUCCESS(Status) (((NTSTATUS)(Status)) >= 0)
+#endif
 
 // We poll for new speed-limit values once every second.
 constexpr base::TimeDelta kSampleInterval = base::Seconds(1);
diff --git a/src/3rdparty/chromium/base/process/memory_win.cc b/src/3rdparty/chromium/base/process/memory_win.cc
index 443bba1..918ca48 100644
--- a/src/3rdparty/chromium/base/process/memory_win.cc
+++ b/src/3rdparty/chromium/base/process/memory_win.cc
@@ -21,13 +21,22 @@
 // It's provided by allocator_shim_win.cc but since that's not always present,
 // we provide a default that falls back to regular malloc.
 typedef void* (*MallocFn)(size_t);
+#if defined(COMPILER_MSVC)
 extern "C" void* (*const malloc_unchecked)(size_t);
+#else
+#if defined(__x86_64__) || defined(__aarch64__)
+extern "C" void* __attribute__((weak, alias("malloc_default"))) (*const malloc_unchecked)(size_t);
+#else
+extern "C" void* __attribute__((weak, alias("_malloc_default"))) (*const malloc_unchecked)(size_t);
+#endif
+#endif
 extern "C" void* (*const malloc_default)(size_t) = &malloc;
 
 #if defined(__clang__)
 #pragma clang diagnostic pop  // -Wglobal-constructors
 #endif
 
+#if defined(COMPILER_MSVC)
 #if defined(_M_IX86)
 #pragma comment(linker, "/alternatename:_malloc_unchecked=_malloc_default")
 #elif defined(_M_X64) || defined(_M_ARM) || defined(_M_ARM64)
@@ -35,11 +44,13 @@ extern "C" void* (*const malloc_default)(size_t) = &malloc;
 #else
 #error Unsupported platform
 #endif
+#endif
 
 namespace base {
 
 namespace {
 
+#if defined(COMPILER_MSVC)
 // Return a non-0 value to retry the allocation.
 int ReleaseReservationOrTerminate(size_t size) {
   constexpr int kRetryAllocation = 1;
@@ -48,6 +59,7 @@ int ReleaseReservationOrTerminate(size_t size) {
   TerminateBecauseOutOfMemory(size);
   return 0;
 }
+#endif
 
 }  // namespace
 
@@ -56,11 +68,13 @@ void EnableTerminationOnHeapCorruption() {
   HeapSetInformation(NULL, HeapEnableTerminationOnCorruption, NULL, 0);
 }
 
+#if defined(COMPILER_MSVC)
 void EnableTerminationOnOutOfMemory() {
   constexpr int kCallNewHandlerOnAllocationFailure = 1;
   _set_new_handler(&ReleaseReservationOrTerminate);
   _set_new_mode(kCallNewHandlerOnAllocationFailure);
 }
+#endif
 
 // Implemented using a weak symbol.
 bool UncheckedMalloc(size_t size, void** result) {
diff --git a/src/3rdparty/chromium/base/strings/string_number_conversions_win.h b/src/3rdparty/chromium/base/strings/string_number_conversions_win.h
index c167e5f..f94142d 100644
--- a/src/3rdparty/chromium/base/strings/string_number_conversions_win.h
+++ b/src/3rdparty/chromium/base/strings/string_number_conversions_win.h
@@ -5,6 +5,7 @@
 #ifndef BASE_STRINGS_STRING_NUMBER_CONVERSIONS_WIN_H_
 #define BASE_STRINGS_STRING_NUMBER_CONVERSIONS_WIN_H_
 
+#include <cstdint>
 #include <string>
 
 #include "base/base_export.h"
diff --git a/src/3rdparty/chromium/base/threading/platform_thread_win.cc b/src/3rdparty/chromium/base/threading/platform_thread_win.cc
index 83a1402..314240f 100644
--- a/src/3rdparty/chromium/base/threading/platform_thread_win.cc
+++ b/src/3rdparty/chromium/base/threading/platform_thread_win.cc
@@ -456,21 +456,28 @@ void SetCurrentThreadQualityOfService(ThreadType thread_type) {
           ::GetModuleHandle(L"kernel32.dll"), "SetThreadInformation"));
   DCHECK(set_thread_information_fn);
 
+#if defined(COMPILER_MSVC)
   bool desire_ecoqos = false;
+#endif
   switch (thread_type) {
     case ThreadType::kBackground:
     case ThreadType::kUtility:
     case ThreadType::kResourceEfficient:
+#if defined(COMPILER_MSVC)
       desire_ecoqos = true;
+#endif
       break;
     case ThreadType::kDefault:
     case ThreadType::kCompositing:
     case ThreadType::kDisplayCritical:
     case ThreadType::kRealtimeAudio:
+#if defined(COMPILER_MSVC)
       desire_ecoqos = false;
+#endif
       break;
   }
 
+#if defined(COMPILER_MSVC)
   THREAD_POWER_THROTTLING_STATE thread_power_throttling_state{
       .Version = THREAD_POWER_THROTTLING_CURRENT_VERSION,
       .ControlMask =
@@ -483,6 +490,7 @@ void SetCurrentThreadQualityOfService(ThreadType thread_type) {
       &thread_power_throttling_state, sizeof(thread_power_throttling_state));
   DPLOG_IF(ERROR, !success)
       << "Failed to set EcoQoS to " << std::boolalpha << desire_ecoqos;
+#endif
 }
 
 }  // namespace
diff --git a/src/3rdparty/chromium/base/trace_event/trace_log.cc b/src/3rdparty/chromium/base/trace_event/trace_log.cc
index 77d12a6..01a7a29 100644
--- a/src/3rdparty/chromium/base/trace_event/trace_log.cc
+++ b/src/3rdparty/chromium/base/trace_event/trace_log.cc
@@ -770,7 +770,7 @@ void TraceLog::UpdateCategoryState(TraceCategory* category) {
     state_flags |= TraceCategory::ENABLED_FOR_RECORDING;
   }
 
-#if BUILDFLAG(IS_WIN)
+#if defined(COMPILER_MSVC)
   if (base::trace_event::TraceEventETWExport::IsCategoryGroupEnabled(
           category->name())) {
     state_flags |= TraceCategory::ENABLED_FOR_ETW_EXPORT;
@@ -1610,7 +1610,7 @@ bool TraceLog::ShouldAddAfterUpdatingState(
     }
   }
 
-#if BUILDFLAG(IS_WIN)
+#if defined(COMPILER_MSVC)
   // This is done sooner rather than later, to avoid creating the event and
   // acquiring the lock, which is not needed for ETW as it's already threadsafe.
   if (*category_group_enabled & TraceCategory::ENABLED_FOR_ETW_EXPORT) {
@@ -1619,7 +1619,7 @@ bool TraceLog::ShouldAddAfterUpdatingState(
     TraceEventETWExport::AddEvent(phase, category_group_enabled, name, id,
                                   args);
   }
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // defined(COMPILER_MSVC)
   return true;
 }
 
@@ -1904,11 +1904,11 @@ void TraceLog::UpdateTraceEventDurationExplicit(
   }
   const AutoReset<bool> resetter(&thread_is_in_trace_event, true);
 
-#if BUILDFLAG(IS_WIN)
+#if defined(COMPILER_MSVC)
   // Generate an ETW event that marks the end of a complete event.
   if (*category_group_enabled & TraceCategory::ENABLED_FOR_ETW_EXPORT)
     TraceEventETWExport::AddCompleteEndEvent(category_group_enabled, name);
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // defined(COMPILER_MSVC)
 
   if (*category_group_enabled & TraceCategory::ENABLED_FOR_RECORDING) {
     auto update_duration_override =
@@ -2166,7 +2166,7 @@ TraceBuffer* TraceLog::CreateTraceBuffer() {
                                : kTraceEventVectorBufferChunks);
 }
 
-#if BUILDFLAG(IS_WIN)
+#if defined(COMPILER_MSVC)
 void TraceLog::UpdateETWCategoryGroupEnabledFlags() {
   // Go through each category and set/clear the ETW bit depending on whether the
   // category is enabled.
@@ -2179,7 +2179,7 @@ void TraceLog::UpdateETWCategoryGroupEnabledFlags() {
     }
   }
 }
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // defined(COMPILER_MSVC)
 
 void TraceLog::SetTraceBufferForTesting(
     std::unique_ptr<TraceBuffer> trace_buffer) {
diff --git a/src/3rdparty/chromium/base/trace_event/trace_logging_minimal_win.h b/src/3rdparty/chromium/base/trace_event/trace_logging_minimal_win.h
index 346e687..d4cdb9b 100644
--- a/src/3rdparty/chromium/base/trace_event/trace_logging_minimal_win.h
+++ b/src/3rdparty/chromium/base/trace_event/trace_logging_minimal_win.h
@@ -224,11 +224,6 @@ class TlmProvider {
   template <class... FieldTys>
   struct DataDescCountSum;  // undefined
 
-  template <>
-  struct DataDescCountSum<> {
-    static constexpr uint8_t value = 0;
-  };
-
   template <class FieldTy1, class... FieldTyRest>
   struct DataDescCountSum<FieldTy1, FieldTyRest...> {
     static constexpr uint8_t value =
@@ -287,6 +282,11 @@ class TlmProvider {
   char provider_metadata_[kMaxProviderMetadataSize] = {};
 };
 
+template <>
+struct TlmProvider::DataDescCountSum<> {
+  static constexpr uint8_t value = 0;
+};
+
 // Base class for field types.
 template <uint8_t data_desc_count,
           uint8_t in_type,
diff --git a/src/3rdparty/chromium/base/win/com_init_balancer.h b/src/3rdparty/chromium/base/win/com_init_balancer.h
index 1f61b3a..2f17e61 100644
--- a/src/3rdparty/chromium/base/win/com_init_balancer.h
+++ b/src/3rdparty/chromium/base/win/com_init_balancer.h
@@ -5,6 +5,7 @@
 #ifndef BASE_WIN_COM_INIT_BALANCER_H_
 #define BASE_WIN_COM_INIT_BALANCER_H_
 
+#if defined(COMPILER_MSVC)
 #include <objidl.h>
 #include <winnt.h>
 #include <wrl/implements.h>
@@ -69,5 +70,6 @@ class BASE_EXPORT ComInitBalancer
 }  // namespace internal
 }  // namespace win
 }  // namespace base
+#endif
 
 #endif  // BASE_WIN_COM_INIT_BALANCER_H_
diff --git a/src/3rdparty/chromium/base/win/core_winrt_util.cc b/src/3rdparty/chromium/base/win/core_winrt_util.cc
index 25f5885..9169ee6 100644
--- a/src/3rdparty/chromium/base/win/core_winrt_util.cc
+++ b/src/3rdparty/chromium/base/win/core_winrt_util.cc
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#define BUILDING_CORE_WINRT_UTIL
+
 #include "base/win/core_winrt_util.h"
 
 namespace base::win {
diff --git a/src/3rdparty/chromium/base/win/core_winrt_util.h b/src/3rdparty/chromium/base/win/core_winrt_util.h
index 3a6e0c9..27bc1b4 100644
--- a/src/3rdparty/chromium/base/win/core_winrt_util.h
+++ b/src/3rdparty/chromium/base/win/core_winrt_util.h
@@ -25,6 +25,7 @@ BASE_EXPORT HRESULT RoGetActivationFactory(HSTRING class_id,
 BASE_EXPORT HRESULT RoActivateInstance(HSTRING class_id,
                                        IInspectable** instance);
 
+#ifndef BUILDING_CORE_WINRT_UTIL
 // Retrieves an activation factory for the type specified.
 template <typename InterfaceType, wchar_t const* runtime_class_id>
 HRESULT GetActivationFactory(InterfaceType** factory) {
@@ -35,6 +36,7 @@ HRESULT GetActivationFactory(InterfaceType** factory) {
   return base::win::RoGetActivationFactory(class_id_hstring.get(),
                                            IID_PPV_ARGS(factory));
 }
+#endif
 
 }  // namespace base::win
 
diff --git a/src/3rdparty/chromium/base/win/propvarutil.h b/src/3rdparty/chromium/base/win/propvarutil.h
index 62a9187..55431b6 100644
--- a/src/3rdparty/chromium/base/win/propvarutil.h
+++ b/src/3rdparty/chromium/base/win/propvarutil.h
@@ -9,7 +9,11 @@
 // Undefine before windows header will make the poisonous defines
 #include "base/win/windows_undefines.inc"
 
+#if defined(COMPILER_MSVC)
 #include <propvarutil.h>  // NOLINT(build/include_order)
+#else
+#include_next <propvarutil.h>
+#endif
 
 // Undefine the poisonous defines
 #include "base/win/windows_undefines.inc"  // NOLINT(build/include)
diff --git a/src/3rdparty/chromium/base/win/scoped_com_initializer.cc b/src/3rdparty/chromium/base/win/scoped_com_initializer.cc
index 83195ec..0dfd95b 100644
--- a/src/3rdparty/chromium/base/win/scoped_com_initializer.cc
+++ b/src/3rdparty/chromium/base/win/scoped_com_initializer.cc
@@ -4,7 +4,9 @@
 
 #include "base/win/scoped_com_initializer.h"
 
+#if defined(COMPILER_MSVC)
 #include <wrl/implements.h>
+#endif
 
 #include <ostream>
 
@@ -25,10 +27,12 @@ ScopedCOMInitializer::ScopedCOMInitializer(SelectMTA mta,
 ScopedCOMInitializer::~ScopedCOMInitializer() {
   DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
   if (Succeeded()) {
+#if defined(COMPILER_MSVC)
     if (com_balancer_) {
       com_balancer_->Disable();
       com_balancer_.Reset();
     }
+#endif
     CoUninitialize();
   }
 }
@@ -38,14 +42,17 @@ bool ScopedCOMInitializer::Succeeded() const {
 }
 
 DWORD ScopedCOMInitializer::GetCOMBalancerReferenceCountForTesting() const {
+#if defined(COMPILER_MSVC)
   if (com_balancer_)
     return com_balancer_->GetReferenceCountForTesting();
+#endif
   return 0;
 }
 
 void ScopedCOMInitializer::Initialize(COINIT init,
                                       Uninitialization uninitialization) {
   DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+#if defined(COMPILER_MSVC)
   // COINIT_DISABLE_OLE1DDE is always added based on:
   // https://docs.microsoft.com/en-us/windows/desktop/learnwin32/initializing-the-com-library
   if (uninitialization == Uninitialization::kBlockPremature) {
@@ -54,6 +61,7 @@ void ScopedCOMInitializer::Initialize(COINIT init,
   }
   hr_ = ::CoInitializeEx(nullptr, init | COINIT_DISABLE_OLE1DDE);
   DCHECK_NE(RPC_E_CHANGED_MODE, hr_) << "Invalid COM thread model change";
+#endif
 }
 
 }  // namespace win
diff --git a/src/3rdparty/chromium/base/win/scoped_com_initializer.h b/src/3rdparty/chromium/base/win/scoped_com_initializer.h
index f86a927..088fdf8 100644
--- a/src/3rdparty/chromium/base/win/scoped_com_initializer.h
+++ b/src/3rdparty/chromium/base/win/scoped_com_initializer.h
@@ -66,7 +66,9 @@ class BASE_EXPORT ScopedCOMInitializer : public ScopedWindowsThreadEnvironment {
   void Initialize(COINIT init, Uninitialization uninitialization);
 
   HRESULT hr_ = S_OK;
+#if defined(COMPILER_MSVC)
   Microsoft::WRL::ComPtr<internal::ComInitBalancer> com_balancer_;
+#endif
   THREAD_CHECKER(thread_checker_);
 };
 
diff --git a/src/3rdparty/chromium/base/win/scoped_variant.cc b/src/3rdparty/chromium/base/win/scoped_variant.cc
index 9f500dc..4a8ed83 100644
--- a/src/3rdparty/chromium/base/win/scoped_variant.cc
+++ b/src/3rdparty/chromium/base/win/scoped_variant.cc
@@ -105,6 +105,7 @@ VARIANT* ScopedVariant::Receive() {
   return &var_;
 }
 
+#if defined(COMPILER_MSVC)
 VARIANT ScopedVariant::Copy() const {
   VARIANT ret = {{{VT_EMPTY}}};
   ::VariantCopy(&ret, &var_);
@@ -177,6 +178,7 @@ int ScopedVariant::Compare(const VARIANT& other, bool ignore_case) const {
   //    the two VARIANTS that share the same VARTYPE.
   return ::VariantCompare(var_, other);
 }
+#endif
 
 void ScopedVariant::Set(const wchar_t* str) {
   DCHECK(!IsLeakableVarType(var_.vt)) << "leaking variant: " << var_.vt;
@@ -285,10 +287,12 @@ void ScopedVariant::Set(SAFEARRAY* array) {
 
 void ScopedVariant::Set(const VARIANT& var) {
   DCHECK(!IsLeakableVarType(var_.vt)) << "leaking variant: " << var_.vt;
+#if defined(COMPILER_MSVC)
   if (FAILED(::VariantCopy(&var_, &var))) {
     DLOG(ERROR) << "VariantCopy failed";
     var_.vt = VT_EMPTY;
   }
+#endif
 }
 
 ScopedVariant& ScopedVariant::operator=(ScopedVariant&& var) {
diff --git a/src/3rdparty/chromium/base/win/shlwapi.h b/src/3rdparty/chromium/base/win/shlwapi.h
index 1579c3d..52e438d 100644
--- a/src/3rdparty/chromium/base/win/shlwapi.h
+++ b/src/3rdparty/chromium/base/win/shlwapi.h
@@ -9,7 +9,11 @@
 // Undefine before windows header will make the poisonous defines
 #include "base/win/windows_undefines.inc"
 
+#if defined(COMPILER_MSVC)
 #include <shlwapi.h>  // NOLINT(build/include_order)
+#else
+#include_next <shlwapi.h>
+#endif
 
 // Undefine the poisonous defines
 #include "base/win/windows_undefines.inc"  // NOLINT(build/include)
diff --git a/src/3rdparty/chromium/base/win/shortcut.cc b/src/3rdparty/chromium/base/win/shortcut.cc
index 38d79a0..f86049b 100644
--- a/src/3rdparty/chromium/base/win/shortcut.cc
+++ b/src/3rdparty/chromium/base/win/shortcut.cc
@@ -169,6 +169,7 @@ bool CreateOrUpdateShortcutLink(const FilePath& shortcut_path,
                                                 properties.app_id.c_str())) {
       return false;
     }
+#if defined(COMPILER_MSVC)
     if (has_dual_mode && !SetBooleanValueForPropertyStore(
                              property_store.Get(), PKEY_AppUserModel_IsDualMode,
                              properties.dual_mode)) {
@@ -180,6 +181,7 @@ bool CreateOrUpdateShortcutLink(const FilePath& shortcut_path,
                                   properties.toast_activator_clsid)) {
       return false;
     }
+#endif
   }
 
   // Release the interfaces to the old shortcut to make sure it doesn't prevent
@@ -297,6 +299,7 @@ bool ResolveShortcutProperties(const FilePath& shortcut_path,
       }
     }
 
+#if defined(COMPILER_MSVC)
     if (options & ShortcutProperties::PROPERTIES_DUAL_MODE) {
       ScopedPropVariant pv_dual_mode;
       if (property_store->GetValue(PKEY_AppUserModel_IsDualMode,
@@ -337,6 +340,7 @@ bool ResolveShortcutProperties(const FilePath& shortcut_path,
           return false;
       }
     }
+#endif
   }
 
   return true;
diff --git a/src/3rdparty/chromium/base/win/sphelper.h b/src/3rdparty/chromium/base/win/sphelper.h
index 4965fc9..a941c4c 100644
--- a/src/3rdparty/chromium/base/win/sphelper.h
+++ b/src/3rdparty/chromium/base/win/sphelper.h
@@ -9,7 +9,11 @@
 // Undefine before windows header will make the poisonous defines
 #include "base/win/windows_undefines.inc"
 
+#if defined(COMPILER_MSVC)
 #include <sphelper.h>  // NOLINT(build/include_order)
+#else
+#include_next <sphelper.h>
+#endif
 
 // Undefine the poisonous defines
 #include "base/win/windows_undefines.inc"  // NOLINT(build/include)
diff --git a/src/3rdparty/chromium/base/win/variant_vector.cc b/src/3rdparty/chromium/base/win/variant_vector.cc
index db2365b..7b66d28 100644
--- a/src/3rdparty/chromium/base/win/variant_vector.cc
+++ b/src/3rdparty/chromium/base/win/variant_vector.cc
@@ -76,6 +76,7 @@ VariantVector::~VariantVector() {
   Reset();
 }
 
+#if defined(COMPILER_MSVC)
 bool VariantVector::operator==(const VariantVector& other) const {
   return !Compare(other);
 }
@@ -83,6 +84,7 @@ bool VariantVector::operator==(const VariantVector& other) const {
 bool VariantVector::operator!=(const VariantVector& other) const {
   return !VariantVector::operator==(other);
 }
+#endif
 
 void VariantVector::Reset() {
   vector_.clear();
@@ -173,6 +175,7 @@ VARIANT VariantVector::ReleaseAsSafearrayVariant() {
   return scoped_variant.Release();
 }
 
+#if defined(COMPILER_MSVC)
 int VariantVector::Compare(const VARIANT& other, bool ignore_case) const {
   // If the element variant types are different, compare against the types.
   if (Type() != (V_VT(&other) & VT_TYPEMASK))
@@ -315,6 +318,7 @@ int VariantVector::Compare(SAFEARRAY* safearray, bool ignore_case) const {
   scoped_safearray.Release();
   return compare_result;
 }
+#endif
 
 template <VARTYPE ElementVartype>
 SAFEARRAY* VariantVector::CreateAndPopulateSafearray() {
diff --git a/src/3rdparty/chromium/base/win/variant_vector.h b/src/3rdparty/chromium/base/win/variant_vector.h
index 8e29728..3468c24 100644
--- a/src/3rdparty/chromium/base/win/variant_vector.h
+++ b/src/3rdparty/chromium/base/win/variant_vector.h
@@ -81,19 +81,6 @@ class BASE_EXPORT VariantVector final {
     vector_.push_back(std::move(scoped_variant));
   }
 
-  // Specialize VT_DATE because ScopedVariant has a separate SetDate method,
-  // this is because VT_R8 and VT_DATE share the same underlying type.
-  template <>
-  void Insert<VT_DATE>(
-      typename internal::VariantConverter<VT_DATE>::Type value) {
-    if (vartype_ == VT_EMPTY)
-      vartype_ = VT_DATE;
-    AssertVartype<VT_DATE>();
-    ScopedVariant scoped_variant;
-    scoped_variant.SetDate(value);
-    vector_.push_back(std::move(scoped_variant));
-  }
-
   // Populates a VARIANT based on what is stored, transferring ownership
   // of managed contents.
   // This is only valid when the VariantVector is empty or has a single element.
@@ -136,6 +123,19 @@ class BASE_EXPORT VariantVector final {
   std::vector<ScopedVariant> vector_;
 };
 
+// Specialize VT_DATE because ScopedVariant has a separate SetDate method,
+// this is because VT_R8 and VT_DATE share the same underlying type.
+template <>
+void VariantVector::Insert<VT_DATE>(
+    typename internal::VariantConverter<VT_DATE>::Type value) {
+  if (vartype_ == VT_EMPTY)
+    vartype_ = VT_DATE;
+  AssertVartype<VT_DATE>();
+  ScopedVariant scoped_variant;
+  scoped_variant.SetDate(value);
+  vector_.push_back(std::move(scoped_variant));
+}
+
 }  // namespace win
 }  // namespace base
 
diff --git a/src/3rdparty/chromium/base/win/vector.h b/src/3rdparty/chromium/base/win/vector.h
index dcbaec7..a5df84a 100644
--- a/src/3rdparty/chromium/base/win/vector.h
+++ b/src/3rdparty/chromium/base/win/vector.h
@@ -5,6 +5,8 @@
 #ifndef BASE_WIN_VECTOR_H_
 #define BASE_WIN_VECTOR_H_
 
+#if defined(COMPILER_MSVC)
+
 #include <ivectorchangedeventargs.h>
 #include <windows.foundation.collections.h>
 #include <wrl/implements.h>
@@ -400,4 +402,6 @@ class Vector
 }  // namespace win
 }  // namespace base
 
+#endif
+
 #endif  // BASE_WIN_VECTOR_H_
diff --git a/src/3rdparty/chromium/base/win/win_util.cc b/src/3rdparty/chromium/base/win/win_util.cc
index 0d81e7c..ef1d859 100644
--- a/src/3rdparty/chromium/base/win/win_util.cc
+++ b/src/3rdparty/chromium/base/win/win_util.cc
@@ -12,7 +12,9 @@
 #include <shobjidl.h>  // Must be before propkey.
 
 #include <inspectable.h>
+#if defined(COMPILER_MSVC)
 #include <mdmregistration.h>
+#endif
 #include <objbase.h>
 #include <propkey.h>
 #include <psapi.h>
@@ -23,8 +25,8 @@
 #include <signal.h>
 #include <stddef.h>
 #include <stdlib.h>
-#include <strsafe.h>
 #include <tchar.h>  // Must be before tpcshrd.h or for any use of _T macro
+#include <strsafe.h> // Must be after tchar.h on MinGW
 #include <tpcshrd.h>
 #include <uiviewsettingsinterop.h>
 #include <windows.ui.viewmanagement.h>
@@ -130,6 +132,7 @@ bool* GetDomainEnrollmentStateStorage() {
   return &state;
 }
 
+#if defined(COMPILER_MSVC)
 bool* GetRegisteredWithManagementStateStorage() {
   static bool state = []() {
     // Mitigate the issues caused by loading DLLs on a background thread
@@ -198,6 +201,7 @@ bool* GetAzureADJoinStateStorage() {
   }();
   return &state;
 }
+#endif
 
 NativeLibrary PinUser32Internal(NativeLibraryLoadError* error) {
   static NativeLibraryLoadError load_error;
@@ -238,6 +242,7 @@ bool IsWindows10OrGreaterTabletMode(HWND hwnd) {
            IsDeviceUsedAsATablet(/*reason=*/nullptr);
   }
 
+#if defined(COMPILER_MSVC)
   ScopedHString view_settings_guid = ScopedHString::Create(
       RuntimeClass_Windows_UI_ViewManagement_UIViewSettings);
   Microsoft::WRL::ComPtr<IUIViewSettingsInterop> view_settings_interop;
@@ -256,6 +261,7 @@ bool IsWindows10OrGreaterTabletMode(HWND hwnd) {
       ABI::Windows::UI::ViewManagement::UserInteractionMode_Mouse;
   view_settings->get_UserInteractionMode(&mode);
   return mode == ABI::Windows::UI::ViewManagement::UserInteractionMode_Touch;
+#endif
 }
 
 // Returns true if a physical keyboard is detected on Windows 8 and up.
@@ -502,7 +508,9 @@ void SetAbortBehaviorForCrashReporting() {
   // Disabling the _CALL_REPORTFAULT behavior is important since otherwise it
   // has the sideffect of clearing our exception filter, which means we
   // don't get any crash.
+#if defined(COMPILER_MSVC) || defined(_UCRT)
   _set_abort_behavior(0, _WRITE_ABORT_MSG | _CALL_REPORTFAULT);
+#endif
 
   // Set a SIGABRT handler for good measure. We will crash even if the default
   // is left in place, however this allows us to crash earlier. And it also
@@ -590,6 +598,7 @@ bool IsEnrolledToDomain() {
   return *GetDomainEnrollmentStateStorage();
 }
 
+#if defined(COMPILER_MSVC)
 bool IsDeviceRegisteredWithManagement() {
   // GetRegisteredWithManagementStateStorage() can be true for devices running
   // the Home sku, however the Home sku does not allow for management of the web
@@ -602,6 +611,7 @@ bool IsDeviceRegisteredWithManagement() {
 bool IsJoinedToAzureAD() {
   return *GetAzureADJoinStateStorage();
 }
+#endif
 
 bool IsUser32AndGdi32Available() {
   static auto is_user32_and_gdi32_available = []() {
@@ -662,6 +672,7 @@ bool GetLoadedModulesSnapshot(HANDLE process, std::vector<HMODULE>* snapshot) {
   return false;
 }
 
+#if defined(COMPILER_MSVC)
 void EnableFlicks(HWND hwnd) {
   ::RemoveProp(hwnd, MICROSOFT_TABLETPENSERVICE_PROPERTY);
 }
@@ -671,6 +682,7 @@ void DisableFlicks(HWND hwnd) {
             reinterpret_cast<HANDLE>(TABLET_DISABLE_FLICKS |
                                      TABLET_DISABLE_FLICKFALLBACKKEYS));
 }
+#endif
 
 void EnableHighDPISupport() {
   if (!IsUser32AndGdi32Available())
@@ -812,6 +824,7 @@ ScopedDomainStateForTesting::~ScopedDomainStateForTesting() {
   *GetDomainEnrollmentStateStorage() = initial_state_;
 }
 
+#if defined(COMPILER_MSVC)
 ScopedDeviceRegisteredWithManagementForTesting::
     ScopedDeviceRegisteredWithManagementForTesting(bool state)
     : initial_state_(IsDeviceRegisteredWithManagement()) {
@@ -829,6 +842,7 @@ ScopedAzureADJoinStateForTesting::ScopedAzureADJoinStateForTesting(bool state)
 ScopedAzureADJoinStateForTesting::~ScopedAzureADJoinStateForTesting() {
   *GetAzureADJoinStateStorage() = initial_state_;
 }
+#endif
 
 }  // namespace win
 }  // namespace base
diff --git a/src/3rdparty/chromium/base/win/windows_types.h b/src/3rdparty/chromium/base/win/windows_types.h
index 67bce28..1e21de6 100644
--- a/src/3rdparty/chromium/base/win/windows_types.h
+++ b/src/3rdparty/chromium/base/win/windows_types.h
@@ -12,6 +12,8 @@
 #include <concurrencysal.h>
 #include <sal.h>
 #include <specstrings.h>
+#include <stdint.h>
+#include <share.h> // for _SH_DENYNO
 
 #include "base/win/win_handle_types.h"
 
@@ -23,8 +25,8 @@ extern "C" {
 
 typedef unsigned long DWORD;  // NOLINT(runtime/int)
 typedef long LONG;            // NOLINT(runtime/int)
-typedef __int64 LONGLONG;
-typedef unsigned __int64 ULONGLONG;
+typedef int64_t LONGLONG;
+typedef uint64_t ULONGLONG;
 
 #define VOID void
 typedef char CHAR;
@@ -33,7 +35,7 @@ typedef long LONG;    // NOLINT(runtime/int)
 typedef int INT;
 typedef unsigned int UINT;
 typedef unsigned int* PUINT;
-typedef unsigned __int64 UINT64;
+typedef uint64_t UINT64;
 typedef void* LPVOID;
 typedef void* PVOID;
 typedef void* HANDLE;
@@ -46,11 +48,11 @@ typedef WORD UWORD;
 typedef WORD ATOM;
 
 #if defined(_WIN64)
-typedef __int64 INT_PTR, *PINT_PTR;
-typedef unsigned __int64 UINT_PTR, *PUINT_PTR;
+typedef int64_t INT_PTR, *PINT_PTR;
+typedef uint64_t UINT_PTR, *PUINT_PTR;
 
-typedef __int64 LONG_PTR, *PLONG_PTR;
-typedef unsigned __int64 ULONG_PTR, *PULONG_PTR;
+typedef int64_t LONG_PTR, *PLONG_PTR;
+typedef uint64_t ULONG_PTR, *PULONG_PTR;
 #else
 typedef __w64 int INT_PTR, *PINT_PTR;
 typedef __w64 unsigned int UINT_PTR, *PUINT_PTR;
@@ -206,11 +208,21 @@ struct CHROME_MSG {
 // Work around there being two slightly different definitions in the SDK.
 #define INVALID_HANDLE_VALUE ((HANDLE)(LONG_PTR)-1)
 #endif
+#ifndef TLS_OUT_OF_INDEXES
 #define TLS_OUT_OF_INDEXES ((DWORD)0xFFFFFFFF)
+#endif
 #define HTNOWHERE 0
 #define MAX_PATH 260
 #define CS_GLOBALCLASS 0x4000
 
+#undef ERROR_SUCCESS
+#undef ERROR_FILE_NOT_FOUND
+#undef ERROR_ACCESS_DENIED
+#undef ERROR_INVALID_HANDLE
+#undef ERROR_SHARING_VIOLATION
+#undef ERROR_LOCK_VIOLATION
+#undef ERROR_MORE_DATA
+
 #define ERROR_SUCCESS 0L
 #define ERROR_FILE_NOT_FOUND 2L
 #define ERROR_ACCESS_DENIED 5L
@@ -218,6 +230,8 @@ struct CHROME_MSG {
 #define ERROR_SHARING_VIOLATION 32L
 #define ERROR_LOCK_VIOLATION 33L
 #define ERROR_MORE_DATA 234L
+
+#if defined(COMPILER_MSVC) || defined(__clang__)
 #define REG_BINARY ( 3ul )
 #define REG_NONE ( 0ul )
 
@@ -232,6 +246,8 @@ struct CHROME_MSG {
 #define HKEY_CLASSES_ROOT (( HKEY ) (ULONG_PTR)((LONG)0x80000000) )
 #define HKEY_LOCAL_MACHINE (( HKEY ) (ULONG_PTR)((LONG)0x80000002) )
 #define HKEY_CURRENT_USER (( HKEY ) (ULONG_PTR)((LONG)0x80000001) )
+#endif
+
 #define KEY_QUERY_VALUE (0x0001)
 #define KEY_SET_VALUE (0x0002)
 #define KEY_CREATE_SUB_KEY (0x0004)
@@ -243,12 +259,16 @@ struct CHROME_MSG {
 #define KEY_WOW64_RES (0x0300)
 
 #define PROCESS_QUERY_INFORMATION (0x0400)
+#if defined(COMPILER_MSVC) || defined(__clang__)
 #define READ_CONTROL (0x00020000L)
 #define SYNCHRONIZE (0x00100000L)
+#endif
 
 #define STANDARD_RIGHTS_READ (READ_CONTROL)
 #define STANDARD_RIGHTS_WRITE (READ_CONTROL)
+#ifndef STANDARD_RIGHTS_ALL
 #define STANDARD_RIGHTS_ALL (0x001F0000L)
+#endif
 
 #define KEY_READ                ((STANDARD_RIGHTS_READ       |\
                                   KEY_QUERY_VALUE            |\
@@ -283,7 +303,9 @@ struct CHROME_MSG {
 
 // Define some macros needed when prototyping Windows functions.
 
+#ifndef DECLSPEC_IMPORT
 #define DECLSPEC_IMPORT __declspec(dllimport)
+#endif
 #define WINBASEAPI DECLSPEC_IMPORT
 #define WINUSERAPI DECLSPEC_IMPORT
 #define WINAPI __stdcall
@@ -291,7 +313,11 @@ struct CHROME_MSG {
 #define CALLBACK __stdcall
 
 // Needed for LockImpl.
+#if defined(COMPILER_MSVC)
 WINBASEAPI _Releases_exclusive_lock_(*SRWLock) VOID WINAPI
+#else
+VOID WINAPI
+#endif
     ReleaseSRWLockExclusive(_Inout_ PSRWLOCK SRWLock);
 WINBASEAPI BOOLEAN WINAPI TryAcquireSRWLockExclusive(_Inout_ PSRWLOCK SRWLock);
 
@@ -357,6 +383,27 @@ inline MSG* ChromeToWindowsType(CHROME_MSG* p) {
 // undefined everywhere. As windows.h is removed from more source files we may
 // be able to shorten this list.
 
+#undef CopyFile
+#undef CreateDirectory
+#undef CreateFile
+#undef CreateService
+#undef DeleteFile
+#undef DispatchMessage
+#undef DrawText
+#undef FindFirstFile
+#undef FindNextFile
+#undef GetClassName
+#undef GetCurrentDirectory
+#undef GetCurrentTime
+#undef GetFileAttributes
+#undef GetMessage
+#undef LoadIcon
+#undef PostMessage
+#undef ReplaceFile
+#undef SendMessage
+#undef SendMessageCallback
+#undef SetCurrentDirectory
+
 #define CopyFile CopyFileW
 #define CreateDirectory CreateDirectoryW
 #define CreateFile CreateFileW
diff --git a/src/3rdparty/chromium/base/win/windows_version.cc b/src/3rdparty/chromium/base/win/windows_version.cc
index 04bd7ae..4ef60dd 100644
--- a/src/3rdparty/chromium/base/win/windows_version.cc
+++ b/src/3rdparty/chromium/base/win/windows_version.cc
@@ -22,7 +22,7 @@
 #include "base/win/registry.h"
 #include "build/build_config.h"
 
-#if !defined(__clang__) && _MSC_FULL_VER < 191125507
+#if !defined(__clang__) && defined(_MSC_VER) && _MSC_FULL_VER < 191125507
 #error VS 2017 Update 3.2 or higher is required
 #endif
 
diff --git a/src/3rdparty/chromium/base/win/wrapped_window_proc.h b/src/3rdparty/chromium/base/win/wrapped_window_proc.h
index 5cacecb..f297ddc 100644
--- a/src/3rdparty/chromium/base/win/wrapped_window_proc.h
+++ b/src/3rdparty/chromium/base/win/wrapped_window_proc.h
@@ -14,6 +14,13 @@
 
 #include "base/base_export.h"
 
+#if defined(__MINGW32__)
+#undef __try
+#define __try try
+#undef __except
+#define __except(x) catch(...)
+#endif
+
 namespace base {
 namespace win {
 
