diff --git a/src/3rdparty/chromium/base/BUILD.gn b/src/3rdparty/chromium/base/BUILD.gn
index c15b2b2..77669b3 100644
--- a/src/3rdparty/chromium/base/BUILD.gn
+++ b/src/3rdparty/chromium/base/BUILD.gn
@@ -110,7 +110,7 @@ config("base_implementation") {
   configs = [ "//build/config/compiler:wexit_time_destructors" ]
 }
 
-if (is_win) {
+if (is_msvc) {
   # This is in a separate config so the flags can be applied to dependents.
   # ldflags in GN aren't automatically inherited.
   config("base_win_linker_flags") {
@@ -129,6 +129,9 @@ if (is_android) {
       "log",  # Used by logging.cc.
     ]
   }
+} else if (is_mingw) {
+  config("base_win_linker_flags") {
+  }
 }
 
 # Base and everything it depends on should be a static library rather than
@@ -1854,6 +1857,20 @@ jumbo_component("base") {
       "strings/string16.cc",
     ]
 
+    if (is_mingw) {
+      sources -= [
+        "win/atl.h",                # requires <atlbase.h>
+        "win/atl_throw.cc",         # requires <atlbase.h>
+        "win/atl_throw.h",
+        "win/enum_variant.cc",      # requires <wrl/implements.h>
+        "win/enum_variant.h",
+        "win/vector.cc",            # requires <wrl/implements.h>
+        "win/vector.h",
+      ]
+
+      defines += [ "STRSAFE_NO_DEPRECATE" ]
+    }
+
     # winternl.h and NTSecAPI.h have different definitions of UNICODE_STRING.
     # There's only one client of NTSecAPI.h in base but several of winternl.h,
     # so exclude the NTSecAPI.h one.
@@ -1877,13 +1894,13 @@ jumbo_component("base") {
     configs += [ "//build/config/compiler:no_size_t_to_int_warning" ]
 
     libs += [
-      "cfgmgr32.lib",
-      "powrprof.lib",
-      "propsys.lib",
-      "setupapi.lib",
-      "userenv.lib",
-      "wbemuuid.lib",
-      "winmm.lib",
+      "cfgmgr32",
+      "powrprof",
+      "propsys",
+      "setupapi",
+      "userenv",
+      "wbemuuid",
+      "winmm",
     ]
     all_dependent_configs += [
       ":base_win_linker_flags",
@@ -2210,7 +2227,7 @@ jumbo_component("base") {
 
     all_dependent_configs += [ "//third_party/perfetto/gn:public_config" ]
 
-    if (is_win) {
+    if (is_msvc) {
       sources += [
         "trace_event/trace_event_etw_export_win.cc",
         "trace_event/trace_event_etw_export_win.h",
@@ -2549,14 +2566,18 @@ if (is_win) {
   # base/test/data/pe_image.
   shared_library("pe_image_test") {
     sources = [ "win/pe_image_test.cc" ]
-    ldflags = [
-      "/DELAYLOAD:cfgmgr32.dll",
-      "/DELAYLOAD:shell32.dll",
-      "/SUBSYSTEM:WINDOWS",
-    ]
+    if (is_mingw) {
+      ldflags = [ "-mwindows" ]
+    } else {
+      ldflags = [
+        "/DELAYLOAD:cfgmgr32.dll",
+        "/DELAYLOAD:shell32.dll",
+        "/SUBSYSTEM:WINDOWS",
+      ]
+    }
     libs = [
-      "cfgmgr32.lib",
-      "shell32.lib",
+      "cfgmgr32",
+      "shell32",
     ]
   }
 
diff --git a/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc_constants.h b/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc_constants.h
index f03ba1e..790ec93 100644
--- a/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc_constants.h
+++ b/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc_constants.h
@@ -195,7 +195,7 @@ NumPartitionPagesPerSuperPage() {
 //
 // __STDCPP_DEFAULT_NEW_ALIGNMENT__ is C++17. As such, it is not defined on all
 // platforms, as Chrome's requirement is C++14 as of 2020.
-#if defined(__STDCPP_DEFAULT_NEW_ALIGNMENT__)
+#if defined(__STDCPP_DEFAULT_NEW_ALIGNMENT__) && !defined(IS_MINGW)
 static constexpr size_t kAlignment =
     std::max(alignof(std::max_align_t), __STDCPP_DEFAULT_NEW_ALIGNMENT__);
 #else
diff --git a/src/3rdparty/chromium/base/containers/id_map.h b/src/3rdparty/chromium/base/containers/id_map.h
index 4caf8a2..4c77f1f 100644
--- a/src/3rdparty/chromium/base/containers/id_map.h
+++ b/src/3rdparty/chromium/base/containers/id_map.h
@@ -153,8 +153,8 @@ class IDMap final {
     }
 
     const Iterator& operator=(const Iterator& iter) {
-      map_ = iter.map;
-      iter_ = iter.iter;
+      map_ = iter.map_;
+      iter_ = iter.iter_;
       Init();
       return *this;
     }
diff --git a/src/3rdparty/chromium/base/export_template.h b/src/3rdparty/chromium/base/export_template.h
index e177fa5..4aac93d 100644
--- a/src/3rdparty/chromium/base/export_template.h
+++ b/src/3rdparty/chromium/base/export_template.h
@@ -138,7 +138,9 @@
 
 EXPORT_TEMPLATE_TEST(DEFAULT, );
 EXPORT_TEMPLATE_TEST(DEFAULT, __attribute__((visibility("default"))));
+#if defined(_MSC_VER)
 EXPORT_TEMPLATE_TEST(EXPORT_DLLEXPORT, __declspec(dllexport));
+#endif
 EXPORT_TEMPLATE_TEST(DEFAULT, __declspec(dllimport));
 
 #undef EXPORT_TEMPLATE_TEST
diff --git a/src/3rdparty/chromium/base/lazy_instance.h b/src/3rdparty/chromium/base/lazy_instance.h
index 94e91e8..37c2219 100644
--- a/src/3rdparty/chromium/base/lazy_instance.h
+++ b/src/3rdparty/chromium/base/lazy_instance.h
@@ -172,7 +172,7 @@ class LazyInstance {
   // MSVC gives a warning that the alignment expands the size of the
   // LazyInstance struct to make the size a multiple of the alignment. This
   // is expected in this case.
-#if defined(OS_WIN)
+#if defined(_MSC_VER)
 #pragma warning(push)
 #pragma warning(disable: 4324)
 #endif
@@ -185,7 +185,7 @@ class LazyInstance {
   // Preallocated space for the Type instance.
   alignas(Type) char private_buf_[sizeof(Type)];
 
-#if defined(OS_WIN)
+#if defined(_MSC_VER)
 #pragma warning(pop)
 #endif
 
diff --git a/src/3rdparty/chromium/base/memory/aligned_memory.cc b/src/3rdparty/chromium/base/memory/aligned_memory.cc
index 7017e31..9ec33e1 100644
--- a/src/3rdparty/chromium/base/memory/aligned_memory.cc
+++ b/src/3rdparty/chromium/base/memory/aligned_memory.cc
@@ -18,7 +18,7 @@ void* AlignedAlloc(size_t size, size_t alignment) {
   DCHECK(bits::IsPowerOfTwo(alignment));
   DCHECK_EQ(alignment % sizeof(void*), 0U);
   void* ptr = nullptr;
-#if defined(COMPILER_MSVC)
+#if defined(OS_WIN)
   ptr = _aligned_malloc(size, alignment);
 #elif defined(OS_ANDROID)
   // Android technically supports posix_memalign(), but does not expose it in
diff --git a/src/3rdparty/chromium/base/memory/aligned_memory.h b/src/3rdparty/chromium/base/memory/aligned_memory.h
index 39a823a..90271aa 100644
--- a/src/3rdparty/chromium/base/memory/aligned_memory.h
+++ b/src/3rdparty/chromium/base/memory/aligned_memory.h
@@ -16,7 +16,7 @@
 #include "base/process/process_metrics.h"
 #include "build/build_config.h"
 
-#if defined(COMPILER_MSVC)
+#if defined(OS_WIN)
 #include <malloc.h>
 #else
 #include <stdlib.h>
@@ -42,7 +42,7 @@ namespace base {
 BASE_EXPORT void* AlignedAlloc(size_t size, size_t alignment);
 
 inline void AlignedFree(void* ptr) {
-#if defined(COMPILER_MSVC)
+#if defined(OS_WIN)
   _aligned_free(ptr);
 #else
   free(ptr);
diff --git a/src/3rdparty/chromium/base/process/memory_win.cc b/src/3rdparty/chromium/base/process/memory_win.cc
index 45e1f01..09fd54e 100644
--- a/src/3rdparty/chromium/base/process/memory_win.cc
+++ b/src/3rdparty/chromium/base/process/memory_win.cc
@@ -7,6 +7,7 @@
 
 #include <windows.h>  // Must be in front of other Windows header files.
 
+#include <memory.h>
 #include <new.h>
 #include <psapi.h>
 #include <stddef.h>
@@ -28,6 +29,9 @@ extern "C" void* (*const malloc_default)(size_t) = &malloc;
 #pragma clang diagnostic pop  // -Wglobal-constructors
 #endif
 
+#if defined(COMPILER_GCC)
+extern "C" void* __attribute__((weak, alias("malloc_default"))) (*const malloc_unchecked)(size_t);
+#else
 #if defined(_M_IX86)
 #pragma comment(linker, "/alternatename:_malloc_unchecked=_malloc_default")
 #elif defined(_M_X64) || defined(_M_ARM) || defined(_M_ARM64)
@@ -35,6 +39,7 @@ extern "C" void* (*const malloc_default)(size_t) = &malloc;
 #else
 #error Unsupported platform
 #endif
+#endif
 
 namespace base {
 
@@ -68,9 +73,13 @@ void EnableTerminationOnHeapCorruption() {
 }
 
 void EnableTerminationOnOutOfMemory() {
+#if defined(COMPILER_GCC)
+  std::set_new_handler(nullptr);
+#else
   constexpr int kCallNewHandlerOnAllocationFailure = 1;
   _set_new_handler(&ReleaseReservationOrTerminate);
   _set_new_mode(kCallNewHandlerOnAllocationFailure);
+#endif
 }
 
 // Implemented using a weak symbol.
diff --git a/src/3rdparty/chromium/base/strings/string_number_conversions_win.h b/src/3rdparty/chromium/base/strings/string_number_conversions_win.h
index 5abcc29..ea12e4c 100644
--- a/src/3rdparty/chromium/base/strings/string_number_conversions_win.h
+++ b/src/3rdparty/chromium/base/strings/string_number_conversions_win.h
@@ -5,6 +5,7 @@
 #ifndef BASE_STRINGS_STRING_NUMBER_CONVERSIONS_WIN_H_
 #define BASE_STRINGS_STRING_NUMBER_CONVERSIONS_WIN_H_
 
+#include <cstdint>
 #include <string>
 
 #include "base/base_export.h"
diff --git a/src/3rdparty/chromium/base/time/time_win.cc b/src/3rdparty/chromium/base/time/time_win.cc
index 36ab0f6..0ae6f23 100644
--- a/src/3rdparty/chromium/base/time/time_win.cc
+++ b/src/3rdparty/chromium/base/time/time_win.cc
@@ -33,8 +33,22 @@
 
 #include "base/time/time.h"
 
+#ifndef _MSC_VER
+#include <windows.h>
+namespace ABI {
+    namespace Windows {
+        namespace Foundation {
+            struct DateTime {
+                INT64 UniversalTime;
+            };
+        }
+    }
+}
+#else
 #include <windows.foundation.h>
 #include <windows.h>
+#endif
+
 #include <mmsystem.h>
 #include <stdint.h>
 
diff --git a/src/3rdparty/chromium/base/trace_event/trace_arguments.h b/src/3rdparty/chromium/base/trace_event/trace_arguments.h
index ca8af5b..7ab270a 100644
--- a/src/3rdparty/chromium/base/trace_event/trace_arguments.h
+++ b/src/3rdparty/chromium/base/trace_event/trace_arguments.h
@@ -636,7 +636,7 @@ class BASE_EXPORT TraceArguments {
 
   // Allow move operations.
   TraceArguments(TraceArguments&& other) noexcept {
-    ::memcpy(this, &other, sizeof(*this));
+    ::memcpy((void*)this, &other, sizeof(*this));
     // All owning pointers were copied to |this|. Setting |other.size_| will
     // mask the pointer values still in |other|.
     other.size_ = 0;
diff --git a/src/3rdparty/chromium/base/trace_event/trace_log.cc b/src/3rdparty/chromium/base/trace_event/trace_log.cc
index 27d132b..f80128b 100644
--- a/src/3rdparty/chromium/base/trace_event/trace_log.cc
+++ b/src/3rdparty/chromium/base/trace_event/trace_log.cc
@@ -49,7 +49,7 @@
 #include "base/trace_event/trace_event.h"
 #include "build/build_config.h"
 
-#if defined(OS_WIN)
+#if defined(IS_MSVC) 
 #include "base/trace_event/trace_event_etw_export_win.h"
 #endif
 
@@ -505,7 +505,7 @@ void TraceLog::UpdateCategoryState(TraceCategory* category) {
     state_flags |= TraceCategory::ENABLED_FOR_RECORDING;
   }
 
-#if defined(OS_WIN)
+#if defined(IS_MSVC)
   if (base::trace_event::TraceEventETWExport::IsCategoryGroupEnabled(
           category->name())) {
     state_flags |= TraceCategory::ENABLED_FOR_ETW_EXPORT;
@@ -1135,7 +1135,7 @@ bool TraceLog::ShouldAddAfterUpdatingState(
     }
   }
 
-#if defined(OS_WIN)
+#if defined(IS_MSVC)
   // This is done sooner rather than later, to avoid creating the event and
   // acquiring the lock, which is not needed for ETW as it's already threadsafe.
   if (*category_group_enabled & TraceCategory::ENABLED_FOR_ETW_EXPORT) {
@@ -1481,7 +1481,7 @@ void TraceLog::UpdateTraceEventDurationExplicit(
     return;
   AutoThreadLocalBoolean thread_is_in_trace_event(&thread_is_in_trace_event_);
 
-#if defined(OS_WIN)
+#if defined(IS_MSVC)
   // Generate an ETW event that marks the end of a complete event.
   if (category_group_enabled_local & TraceCategory::ENABLED_FOR_ETW_EXPORT)
     TraceEventETWExport::AddCompleteEndEvent(category_group_enabled, name);
@@ -1746,7 +1746,7 @@ TraceBuffer* TraceLog::CreateTraceBuffer() {
                                : kTraceEventVectorBufferChunks);
 }
 
-#if defined(OS_WIN)
+#if defined(IS_MSVC)
 void TraceLog::UpdateETWCategoryGroupEnabledFlags() {
   // Go through each category and set/clear the ETW bit depending on whether the
   // category is enabled.
diff --git a/src/3rdparty/chromium/base/trace_event/trace_logging_minimal_win.h b/src/3rdparty/chromium/base/trace_event/trace_logging_minimal_win.h
index e7b640e..cfbb7a5 100644
--- a/src/3rdparty/chromium/base/trace_event/trace_logging_minimal_win.h
+++ b/src/3rdparty/chromium/base/trace_event/trace_logging_minimal_win.h
@@ -222,11 +222,6 @@ class TlmProvider {
   template <class... FieldTys>
   struct DataDescCountSum;  // undefined
 
-  template <>
-  struct DataDescCountSum<> {
-    static constexpr uint8_t value = 0;
-  };
-
   template <class FieldTy1, class... FieldTyRest>
   struct DataDescCountSum<FieldTy1, FieldTyRest...> {
     static constexpr uint8_t value =
@@ -285,6 +280,11 @@ class TlmProvider {
   char provider_metadata_[kMaxProviderMetadataSize] = {};
 };
 
+template <>
+struct TlmProvider::DataDescCountSum<> {
+  static constexpr uint8_t value = 0;
+};
+
 // Base class for field types.
 template <uint8_t data_desc_count,
           uint8_t in_type,
diff --git a/src/3rdparty/chromium/base/win/com_init_balancer.h b/src/3rdparty/chromium/base/win/com_init_balancer.h
index cf7860c..68b7bfc 100644
--- a/src/3rdparty/chromium/base/win/com_init_balancer.h
+++ b/src/3rdparty/chromium/base/win/com_init_balancer.h
@@ -7,7 +7,9 @@
 
 #include <objidl.h>
 #include <winnt.h>
+#if defined(COMPILER_MSVC)
 #include <wrl/implements.h>
+#endif
 
 #include "base/base_export.h"
 #include "base/optional.h"
@@ -27,9 +29,13 @@ namespace internal {
 // revoke the registered spy and allow for the successful uninitialization of
 // the COM library.
 class BASE_EXPORT ComInitBalancer
+#if defined(COMPILER_MSVC)
     : public Microsoft::WRL::RuntimeClass<
           Microsoft::WRL::RuntimeClassFlags<Microsoft::WRL::ClassicCom>,
           IInitializeSpy> {
+#else
+    : public IInitializeSpy {
+#endif
  public:
   // Constructs a COM initialize balancer. |co_init| defines the apartment's
   // concurrency model used by the balancer.
@@ -38,7 +44,7 @@ class BASE_EXPORT ComInitBalancer
   ComInitBalancer(const ComInitBalancer&) = delete;
   ComInitBalancer& operator=(const ComInitBalancer&) = delete;
 
-  ~ComInitBalancer() override;
+  ~ComInitBalancer();
 
   // Disables balancer by revoking the registered spy and consequently
   // unblocking attempts to uninitialize the COM library.
@@ -49,12 +55,12 @@ class BASE_EXPORT ComInitBalancer
  private:
   // IInitializeSpy:
   IFACEMETHODIMP PreInitialize(DWORD apartment_type,
-                               DWORD reference_count) override;
+                               DWORD reference_count);
   IFACEMETHODIMP PostInitialize(HRESULT result,
                                 DWORD apartment_type,
-                                DWORD new_reference_count) override;
-  IFACEMETHODIMP PreUninitialize(DWORD reference_count) override;
-  IFACEMETHODIMP PostUninitialize(DWORD new_reference_count) override;
+                                DWORD new_reference_count);
+  IFACEMETHODIMP PreUninitialize(DWORD reference_count);
+  IFACEMETHODIMP PostUninitialize(DWORD new_reference_count);
 
   const DWORD co_init_;
 
diff --git a/src/3rdparty/chromium/base/win/pe_image.cc b/src/3rdparty/chromium/base/win/pe_image.cc
index 817e5c5..3d52964 100644
--- a/src/3rdparty/chromium/base/win/pe_image.cc
+++ b/src/3rdparty/chromium/base/win/pe_image.cc
@@ -7,7 +7,6 @@
 
 #include "base/win/pe_image.h"
 
-#include <delayimp.h>
 #include <stddef.h>
 #include <set>
 #include <string>
diff --git a/src/3rdparty/chromium/base/win/pe_image.h b/src/3rdparty/chromium/base/win/pe_image.h
index 97a4d30..e9609e5 100644
--- a/src/3rdparty/chromium/base/win/pe_image.h
+++ b/src/3rdparty/chromium/base/win/pe_image.h
@@ -19,7 +19,7 @@
 // The Windows 8 SDK defines FACILITY_VISUALCPP in winerror.h.
 #undef FACILITY_VISUALCPP
 #endif
-#include <DelayIMP.h>
+#include <delayimp.h>
 
 namespace base {
 namespace win {
diff --git a/src/3rdparty/chromium/base/win/propvarutil.h b/src/3rdparty/chromium/base/win/propvarutil.h
index d95c5ad..e1c7a40 100644
--- a/src/3rdparty/chromium/base/win/propvarutil.h
+++ b/src/3rdparty/chromium/base/win/propvarutil.h
@@ -9,7 +9,12 @@
 // Undefine before windows header will make the poisonous defines
 #include "base/win/windows_undefines.inc"
 
+#ifdef _MSC_VER
 #include <propvarutil.h>
+#else
+#define _PROPSYS_
+#include_next <propvarutil.h>
+#endif
 
 // Undefine the poisonous defines
 #include "base/win/windows_undefines.inc"
diff --git a/src/3rdparty/chromium/base/win/scoped_com_initializer.cc b/src/3rdparty/chromium/base/win/scoped_com_initializer.cc
index 6d80ff4..9ef7883 100644
--- a/src/3rdparty/chromium/base/win/scoped_com_initializer.cc
+++ b/src/3rdparty/chromium/base/win/scoped_com_initializer.cc
@@ -4,7 +4,9 @@
 
 #include "base/win/scoped_com_initializer.h"
 
+#if defined(COMPILER_MSVC)
 #include <wrl/implements.h>
+#endif
 
 #include "base/check_op.h"
 
@@ -44,6 +46,7 @@ DWORD ScopedCOMInitializer::GetCOMBalancerReferenceCountForTesting() const {
 void ScopedCOMInitializer::Initialize(COINIT init,
                                       Uninitialization uninitialization) {
   DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+#if defined(COMPILER_MSVC)
   // COINIT_DISABLE_OLE1DDE is always added based on:
   // https://docs.microsoft.com/en-us/windows/desktop/learnwin32/initializing-the-com-library
   if (uninitialization == Uninitialization::kBlockPremature) {
@@ -52,6 +55,7 @@ void ScopedCOMInitializer::Initialize(COINIT init,
   }
   hr_ = ::CoInitializeEx(nullptr, init | COINIT_DISABLE_OLE1DDE);
   DCHECK_NE(RPC_E_CHANGED_MODE, hr_) << "Invalid COM thread model change";
+#endif
 }
 
 }  // namespace win
diff --git a/src/3rdparty/chromium/base/win/scoped_variant.cc b/src/3rdparty/chromium/base/win/scoped_variant.cc
index 352c22b..823f497 100644
--- a/src/3rdparty/chromium/base/win/scoped_variant.cc
+++ b/src/3rdparty/chromium/base/win/scoped_variant.cc
@@ -108,7 +108,7 @@ VARIANT* ScopedVariant::Receive() {
 
 VARIANT ScopedVariant::Copy() const {
   VARIANT ret = {{{VT_EMPTY}}};
-  ::VariantCopy(&ret, &var_);
+  ::VariantCopy(&ret, const_cast<VARIANT*>(&var_)); // FIXME: VariantCopy is implemented differently
   return ret;
 }
 
@@ -174,9 +174,11 @@ int ScopedVariant::Compare(const VARIANT& other, bool ignore_case) const {
     }
   }
 
+#if !defined(COMPILER_GCC) // FIXME: missing VariantCompare in <propvarutil.h>
   // 5. Otherwise returns the lexicographical comparison of the values held by
   //    the two VARIANTS that share the same VARTYPE.
   return ::VariantCompare(var_, other);
+#endif
 }
 
 void ScopedVariant::Set(const wchar_t* str) {
@@ -286,7 +288,7 @@ void ScopedVariant::Set(SAFEARRAY* array) {
 
 void ScopedVariant::Set(const VARIANT& var) {
   DCHECK(!IsLeakableVarType(var_.vt)) << "leaking variant: " << var_.vt;
-  if (FAILED(::VariantCopy(&var_, &var))) {
+  if (FAILED(::VariantCopy(&var_, const_cast<VARIANT*>(&var)))) {
     DLOG(ERROR) << "VariantCopy failed";
     var_.vt = VT_EMPTY;
   }
diff --git a/src/3rdparty/chromium/base/win/shlwapi.h b/src/3rdparty/chromium/base/win/shlwapi.h
index b60353c..d5ca1de 100644
--- a/src/3rdparty/chromium/base/win/shlwapi.h
+++ b/src/3rdparty/chromium/base/win/shlwapi.h
@@ -9,7 +9,11 @@
 // Undefine before windows header will make the poisonous defines
 #include "base/win/windows_undefines.inc"
 
+#ifdef _MSC_VER
 #include <shlwapi.h>
+#else
+#include_next <shlwapi.h>
+#endif
 
 // Undefine the poisonous defines
 #include "base/win/windows_undefines.inc"
diff --git a/src/3rdparty/chromium/base/win/shortcut.cc b/src/3rdparty/chromium/base/win/shortcut.cc
index a9d8ece..7a75b24 100644
--- a/src/3rdparty/chromium/base/win/shortcut.cc
+++ b/src/3rdparty/chromium/base/win/shortcut.cc
@@ -156,6 +156,7 @@ bool CreateOrUpdateShortcutLink(const FilePath& shortcut_path,
                                                 properties.app_id.c_str())) {
       return false;
     }
+#if !defined(COMPILER_GCC) // FIXME: missing PKEY_AppUserModel_IsDualMode, PKEY_AppUserModel_ToastActivatorCLSID
     if (has_dual_mode && !SetBooleanValueForPropertyStore(
                              property_store.Get(), PKEY_AppUserModel_IsDualMode,
                              properties.dual_mode)) {
@@ -167,6 +168,7 @@ bool CreateOrUpdateShortcutLink(const FilePath& shortcut_path,
                                   properties.toast_activator_clsid)) {
       return false;
     }
+#endif
   }
 
   // Release the interfaces to the old shortcut to make sure it doesn't prevent
@@ -289,6 +291,7 @@ bool ResolveShortcutProperties(const FilePath& shortcut_path,
       }
     }
 
+#if !defined(COMPILER_GCC) // FIXME: missing PKEY_AppUserModel_IsDualMode, PKEY_AppUserModel_ToastActivatorCLSID
     if (options & ShortcutProperties::PROPERTIES_DUAL_MODE) {
       ScopedPropVariant pv_dual_mode;
       if (property_store->GetValue(PKEY_AppUserModel_IsDualMode,
@@ -329,6 +332,7 @@ bool ResolveShortcutProperties(const FilePath& shortcut_path,
           return false;
       }
     }
+#endif
   }
 
   return true;
diff --git a/src/3rdparty/chromium/base/win/sphelper.h b/src/3rdparty/chromium/base/win/sphelper.h
index 38a547f..f115668 100644
--- a/src/3rdparty/chromium/base/win/sphelper.h
+++ b/src/3rdparty/chromium/base/win/sphelper.h
@@ -9,7 +9,11 @@
 // Undefine before windows header will make the poisonous defines
 #include "base/win/windows_undefines.inc"
 
+#ifdef _MSC_VER
 #include <sphelper.h>
+#else
+#include_next <sphelper.h>
+#endif
 
 // Undefine the poisonous defines
 #include "base/win/windows_undefines.inc"
diff --git a/src/3rdparty/chromium/base/win/variant_vector.h b/src/3rdparty/chromium/base/win/variant_vector.h
index a65e39d..d807ebb 100644
--- a/src/3rdparty/chromium/base/win/variant_vector.h
+++ b/src/3rdparty/chromium/base/win/variant_vector.h
@@ -81,18 +81,6 @@ class BASE_EXPORT VariantVector final {
     vector_.push_back(std::move(scoped_variant));
   }
 
-  // Specialize VT_DATE because ScopedVariant has a separate SetDate method,
-  // this is because VT_R8 and VT_DATE share the same underlying type.
-  template <>
-  void Insert<VT_DATE>(typename internal::VariantUtil<VT_DATE>::Type value) {
-    if (vartype_ == VT_EMPTY)
-      vartype_ = VT_DATE;
-    AssertVartype<VT_DATE>();
-    ScopedVariant scoped_variant;
-    scoped_variant.SetDate(value);
-    vector_.push_back(std::move(scoped_variant));
-  }
-
   // Populates a VARIANT based on what is stored, transferring ownership
   // of managed contents.
   // This is only valid when the VariantVector is empty or has a single element.
@@ -134,6 +122,18 @@ class BASE_EXPORT VariantVector final {
   std::vector<ScopedVariant> vector_;
 };
 
+// Specialize VT_DATE because ScopedVariant has a separate SetDate method,
+// this is because VT_R8 and VT_DATE share the same underlying type.
+template <>
+void VariantVector::Insert<VT_DATE>(typename internal::VariantUtil<VT_DATE>::Type value) {
+  if (vartype_ == VT_EMPTY)
+    vartype_ = VT_DATE;
+  AssertVartype<VT_DATE>();
+  ScopedVariant scoped_variant;
+  scoped_variant.SetDate(value);
+  vector_.push_back(std::move(scoped_variant));
+}
+
 }  // namespace win
 }  // namespace base
 
diff --git a/src/3rdparty/chromium/base/win/vector.h b/src/3rdparty/chromium/base/win/vector.h
index 9773415..c361cb6 100644
--- a/src/3rdparty/chromium/base/win/vector.h
+++ b/src/3rdparty/chromium/base/win/vector.h
@@ -5,6 +5,7 @@
 #ifndef BASE_WIN_VECTOR_H_
 #define BASE_WIN_VECTOR_H_
 
+#if defined(COMPILER_MSVC)
 #include <ivectorchangedeventargs.h>
 #include <windows.foundation.collections.h>
 #include <wrl/implements.h>
@@ -400,4 +401,6 @@ class Vector
 }  // namespace win
 }  // namespace base
 
+#endif // COMPILER_MSVC
+
 #endif  // BASE_WIN_VECTOR_H_
diff --git a/src/3rdparty/chromium/base/win/win_util.cc b/src/3rdparty/chromium/base/win/win_util.cc
index 6101564..287770b 100644
--- a/src/3rdparty/chromium/base/win/win_util.cc
+++ b/src/3rdparty/chromium/base/win/win_util.cc
@@ -11,7 +11,9 @@
 #include <shobjidl.h>  // Must be before propkey.
 
 #include <inspectable.h>
+#if defined(COMPILER_MSVC)
 #include <mdmregistration.h>
+#endif
 #include <objbase.h>
 #include <propkey.h>
 #include <psapi.h>
@@ -22,8 +24,8 @@
 #include <signal.h>
 #include <stddef.h>
 #include <stdlib.h>
-#include <strsafe.h>
 #include <tchar.h>  // Must be before tpcshrd.h or for any use of _T macro
+#include <strsafe.h> // Must be after tchar.h on MinGW
 #include <tpcshrd.h>
 #include <uiviewsettingsinterop.h>
 #include <windows.ui.viewmanagement.h>
@@ -154,6 +156,7 @@ bool* GetDomainEnrollmentStateStorage() {
   return &state;
 }
 
+#if defined(COMPILER_MSVC)
 bool* GetRegisteredWithManagementStateStorage() {
   static bool state = []() {
     // Mitigate the issues caused by loading DLLs on a background thread
@@ -182,6 +185,7 @@ bool* GetRegisteredWithManagementStateStorage() {
 
   return &state;
 }
+#endif
 
 NativeLibrary PinUser32Internal(NativeLibraryLoadError* error) {
   static NativeLibraryLoadError load_error;
@@ -208,6 +212,7 @@ bool IsWindows10TabletMode(HWND hwnd) {
     return false;
   }
 
+#if defined(COMPILER_MSVC)
   ScopedHString view_settings_guid = ScopedHString::Create(
       RuntimeClass_Windows_UI_ViewManagement_UIViewSettings);
   Microsoft::WRL::ComPtr<IUIViewSettingsInterop> view_settings_interop;
@@ -226,6 +231,7 @@ bool IsWindows10TabletMode(HWND hwnd) {
       ABI::Windows::UI::ViewManagement::UserInteractionMode_Mouse;
   view_settings->get_UserInteractionMode(&mode);
   return mode == ABI::Windows::UI::ViewManagement::UserInteractionMode_Touch;
+#endif
 }
 
 // Returns true if a physical keyboard is detected on Windows 8 and up.
@@ -495,7 +501,9 @@ void SetAbortBehaviorForCrashReporting() {
   // Disabling the _CALL_REPORTFAULT behavior is important since otherwise it
   // has the sideffect of clearing our exception filter, which means we
   // don't get any crash.
+#if defined(COMPILER_MSVC) || (defined(COMPILER_GCC) && defined(_UCRT))
   _set_abort_behavior(0, _WRITE_ABORT_MSG | _CALL_REPORTFAULT);
+#endif
 
   // Set a SIGABRT handler for good measure. We will crash even if the default
   // is left in place, however this allows us to crash earlier. And it also
@@ -588,9 +596,11 @@ bool IsEnrolledToDomain() {
   return *GetDomainEnrollmentStateStorage();
 }
 
+#if defined(COMPILER_MSVC)
 bool IsDeviceRegisteredWithManagement() {
   return *GetRegisteredWithManagementStateStorage();
 }
+#endif
 
 bool IsUser32AndGdi32Available() {
   static auto is_user32_and_gdi32_available = []() {
@@ -664,6 +674,7 @@ bool GetLoadedModulesSnapshot(HANDLE process, std::vector<HMODULE>* snapshot) {
   return false;
 }
 
+#if defined(COMPILER_MSVC)
 void EnableFlicks(HWND hwnd) {
   ::RemoveProp(hwnd, MICROSOFT_TABLETPENSERVICE_PROPERTY);
 }
@@ -673,6 +684,7 @@ void DisableFlicks(HWND hwnd) {
             reinterpret_cast<HANDLE>(TABLET_DISABLE_FLICKS |
                                      TABLET_DISABLE_FLICKFALLBACKKEYS));
 }
+#endif
 
 bool IsProcessPerMonitorDpiAware() {
   enum class PerMonitorDpiAware {
@@ -817,6 +829,7 @@ ScopedDomainStateForTesting::~ScopedDomainStateForTesting() {
   *GetDomainEnrollmentStateStorage() = initial_state_;
 }
 
+#if defined(COMPILER_MSVC)
 ScopedDeviceRegisteredWithManagementForTesting::
     ScopedDeviceRegisteredWithManagementForTesting(bool state)
     : initial_state_(IsDeviceRegisteredWithManagement()) {
@@ -827,6 +840,7 @@ ScopedDeviceRegisteredWithManagementForTesting::
     ~ScopedDeviceRegisteredWithManagementForTesting() {
   *GetRegisteredWithManagementStateStorage() = initial_state_;
 }
+#endif
 
 }  // namespace win
 }  // namespace base
diff --git a/src/3rdparty/chromium/base/win/windows_types.h b/src/3rdparty/chromium/base/win/windows_types.h
index d1f3e8d..4868172 100644
--- a/src/3rdparty/chromium/base/win/windows_types.h
+++ b/src/3rdparty/chromium/base/win/windows_types.h
@@ -12,6 +12,8 @@
 #include <concurrencysal.h>
 #include <sal.h>
 #include <specstrings.h>
+#include <share.h> // for _SH_DENYNO
+#include <stdint.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -21,8 +23,9 @@ extern "C" {
 
 typedef unsigned long DWORD;
 typedef long LONG;
-typedef __int64 LONGLONG;
-typedef unsigned __int64 ULONGLONG;
+typedef int64_t LONGLONG;
+typedef uint64_t ULONGLONG;
+
 
 #define VOID void
 typedef char CHAR;
@@ -43,11 +46,11 @@ typedef WORD UWORD;
 typedef WORD ATOM;
 
 #if defined(_WIN64)
-typedef __int64 INT_PTR, *PINT_PTR;
-typedef unsigned __int64 UINT_PTR, *PUINT_PTR;
+typedef int64_t INT_PTR, *PINT_PTR;
+typedef uint64_t UINT_PTR, *PUINT_PTR;
 
-typedef __int64 LONG_PTR, *PLONG_PTR;
-typedef unsigned __int64 ULONG_PTR, *PULONG_PTR;
+typedef int64_t LONG_PTR, *PLONG_PTR;
+typedef uint64_t ULONG_PTR, *PULONG_PTR;
 #else
 typedef __w64 int INT_PTR, *PINT_PTR;
 typedef __w64 unsigned int UINT_PTR, *PUINT_PTR;
@@ -144,11 +147,28 @@ struct CHROME_CONDITION_VARIABLE {
 // Work around there being two slightly different definitions in the SDK.
 #define INVALID_HANDLE_VALUE ((HANDLE)(LONG_PTR)-1)
 #endif
+#ifndef TLS_OUT_OF_INDEXES
 #define TLS_OUT_OF_INDEXES ((DWORD)0xFFFFFFFF)
+#endif
 #define HTNOWHERE 0
 #define MAX_PATH 260
 #define CS_GLOBALCLASS 0x4000
 
+#ifndef _MSC_VER
+#undef ERROR_SUCCESS
+#undef ERROR_FILE_NOT_FOUND
+#undef ERROR_ACCESS_DENIED
+#undef ERROR_INVALID_HANDLE
+#undef ERROR_SHARING_VIOLATION
+#undef ERROR_LOCK_VIOLATION
+#undef REG_BINARY
+#undef READ_CONTROL
+#undef SYNCHRONIZE
+#undef HKEY_CLASSES_ROOT
+#undef HKEY_LOCAL_MACHINE
+#undef HKEY_CURRENT_USER
+#endif
+
 #define ERROR_SUCCESS 0L
 #define ERROR_FILE_NOT_FOUND 2L
 #define ERROR_ACCESS_DENIED 5L
@@ -157,10 +177,16 @@ struct CHROME_CONDITION_VARIABLE {
 #define ERROR_LOCK_VIOLATION 33L
 #define REG_BINARY ( 3ul )
 
+#ifndef STATUS_PENDING
 #define STATUS_PENDING ((DWORD   )0x00000103L)
+#endif
 #define STILL_ACTIVE STATUS_PENDING
+#ifndef SUCCEEDED
 #define SUCCEEDED(hr) (((HRESULT)(hr)) >= 0)
+#endif
+#ifndef FAILED
 #define FAILED(hr) (((HRESULT)(hr)) < 0)
+#endif
 
 #define HKEY_CLASSES_ROOT (( HKEY ) (ULONG_PTR)((LONG)0x80000000) )
 #define HKEY_LOCAL_MACHINE (( HKEY ) (ULONG_PTR)((LONG)0x80000002) )
@@ -180,7 +206,9 @@ struct CHROME_CONDITION_VARIABLE {
 
 #define STANDARD_RIGHTS_READ (READ_CONTROL)
 #define STANDARD_RIGHTS_WRITE (READ_CONTROL)
+#ifndef STANDARD_RIGHTS_ALL
 #define STANDARD_RIGHTS_ALL (0x001F0000L)
+#endif
 
 #define KEY_READ                ((STANDARD_RIGHTS_READ       |\
                                   KEY_QUERY_VALUE            |\
@@ -210,14 +238,22 @@ struct CHROME_CONDITION_VARIABLE {
 
 // Define some macros needed when prototyping Windows functions.
 
+#ifndef DECLSPEC_IMPORT
 #define DECLSPEC_IMPORT __declspec(dllimport)
+#endif
 #define WINBASEAPI DECLSPEC_IMPORT
 #define WINUSERAPI DECLSPEC_IMPORT
 #define WINAPI __stdcall
+#ifndef CALLBACK
 #define CALLBACK __stdcall
+#endif
 
 // Needed for LockImpl.
+#ifdef _MSC_VER
 WINBASEAPI _Releases_exclusive_lock_(*SRWLock) VOID WINAPI
+#else
+VOID WINAPI
+#endif
     ReleaseSRWLockExclusive(_Inout_ PSRWLOCK SRWLock);
 WINBASEAPI BOOLEAN WINAPI TryAcquireSRWLockExclusive(_Inout_ PSRWLOCK SRWLock);
 
@@ -246,6 +282,36 @@ WINBASEAPI VOID WINAPI SetLastError(_In_ DWORD dwErrCode);
 // ensures that the same renaming will happen everywhere. Includes of this file
 // can be added wherever needed to ensure this consistent renaming.
 
+#ifndef _MSC_VER
+#undef CopyFile
+#undef CreateDirectory
+#undef CreateEvent
+#undef CreateFile
+#undef CreateService
+#undef DeleteFile
+#undef DispatchMessage
+#undef DrawText
+#undef FindFirstFile
+#undef FindNextFile
+#undef GetComputerName
+#undef GetCurrentDirectory
+#undef GetCurrentTime
+#undef GetFileAttributes
+#undef GetMessage
+#undef GetUserName
+#undef LoadIcon
+#undef LoadImage
+#undef PostMessage
+#undef RemoveDirectory
+#undef ReplaceFile
+#undef ReportEvent
+#undef SendMessage
+#undef SendMessageCallback
+#undef SetCurrentDirectory
+#undef StartService
+#undef UpdateResource
+#endif
+
 #define CopyFile CopyFileW
 #define CreateDirectory CreateDirectoryW
 #define CreateEvent CreateEventW
diff --git a/src/3rdparty/chromium/base/win/windows_version.cc b/src/3rdparty/chromium/base/win/windows_version.cc
index 5010309..0d52d9e 100644
--- a/src/3rdparty/chromium/base/win/windows_version.cc
+++ b/src/3rdparty/chromium/base/win/windows_version.cc
@@ -19,7 +19,7 @@
 #include "base/strings/utf_string_conversions.h"
 #include "base/win/registry.h"
 
-#if !defined(__clang__) && _MSC_FULL_VER < 191125507
+#if !defined(__clang__) && defined(_MSC_VER) && _MSC_FULL_VER < 191125507
 #error VS 2017 Update 3.2 or higher is required
 #endif
 
diff --git a/src/3rdparty/chromium/base/win/wrapped_window_proc.h b/src/3rdparty/chromium/base/win/wrapped_window_proc.h
index 79dff1e..86928e9 100644
--- a/src/3rdparty/chromium/base/win/wrapped_window_proc.h
+++ b/src/3rdparty/chromium/base/win/wrapped_window_proc.h
@@ -14,6 +14,13 @@
 
 #include "base/base_export.h"
 
+#ifndef _MSC_VER
+#undef __try
+#define __try try
+#undef __except
+#define __except(x) catch(...)
+#endif
+
 namespace base {
 namespace win {
 
