diff --git a/src/3rdparty/chromium/third_party/perfetto/gn/BUILD.gn b/src/3rdparty/chromium/third_party/perfetto/gn/BUILD.gn
index fc5dcd8..85058ed 100644
--- a/src/3rdparty/chromium/third_party/perfetto/gn/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/perfetto/gn/BUILD.gn
@@ -271,7 +271,7 @@ config("protobuf_gen_config") {
       # Using -isystem instead of include_dirs (-I), so we don't need to
       # suppress warnings coming from libprotobuf headers. Doing so would mask
       # warnings in our own code.
-      "-isystem",
+      "-I",
       rebase_path("../buildtools/protobuf/src", root_build_dir),
     ]
   }
diff --git a/src/3rdparty/chromium/third_party/perfetto/gn/standalone/BUILD.gn b/src/3rdparty/chromium/third_party/perfetto/gn/standalone/BUILD.gn
index 3938a89..0425a0d 100644
--- a/src/3rdparty/chromium/third_party/perfetto/gn/standalone/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/perfetto/gn/standalone/BUILD.gn
@@ -95,16 +95,23 @@ config("default") {
   cflags += [
     "-fstrict-aliasing",
     "-fstack-protector-strong",
-    "-fPIC",
-    "-g",
-    "-Wformat",
   ]
 
+  if (is_mingw) {
+    cflags += [ "-g" ]
+  } else {
+    cflags += [
+      "-fPIC",
+      "-g",
+      "-Wformat",
+    ]
+  }
+
   if (!is_fuzzer) {
     cflags += [ "-Werror" ]
   }
 
-  if (is_clang) {
+  if (is_clang && !is_mingw) {
     cflags += [
       # Color compiler output, see https://github.com/ninja-build/ninja/wiki/FAQ
       "-fcolor-diagnostics",
@@ -140,7 +147,7 @@ config("default") {
       "-m32",
       "-lgcc",
     ]
-  } else if (current_cpu == "arm64") {
+  } else if (current_cpu == "x64" || current_cpu == "arm64") {
     cflags += [ "-fno-omit-frame-pointer" ]
   }
 
@@ -218,7 +225,7 @@ config("release") {
     "-fdata-sections",
     "-ffunction-sections",
   ]
-  if (is_android) {
+  if (is_android || is_mingw) {
     cflags += [ "-O2" ]
   } else if (is_fuzzer) {
     cflags += [ "-O1" ]
@@ -227,6 +234,8 @@ config("release") {
   }
   if (is_mac) {
     ldflags = [ "-dead_strip" ]
+  } else if (is_mingw) {
+    ldflags = [ "-Wl,--gc-sections" ]
   } else {
     ldflags = [
       "-Wl,--gc-sections",
diff --git a/src/3rdparty/chromium/third_party/perfetto/gn/standalone/BUILDCONFIG.gn b/src/3rdparty/chromium/third_party/perfetto/gn/standalone/BUILDCONFIG.gn
index df49ef7..d2f1c79 100644
--- a/src/3rdparty/chromium/third_party/perfetto/gn/standalone/BUILDCONFIG.gn
+++ b/src/3rdparty/chromium/third_party/perfetto/gn/standalone/BUILDCONFIG.gn
@@ -36,10 +36,10 @@ is_linux = current_os == "linux"
 is_linux_host = host_os == "linux"
 is_mac = current_os == "mac"
 is_mac_host = host_os == "mac"
+is_win = current_os == "win"
 
 # Building with Windows/Fuchsia/nacl is currently only supported in the Chromium
 # tree so always set this to false.
-is_win = false
 is_fuchsia = false
 is_nacl = false
 
diff --git a/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/optional.h b/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/optional.h
index b093510..da5a9e2 100644
--- a/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/optional.h
+++ b/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/optional.h
@@ -381,7 +381,7 @@ using RemoveCvRefT =
 // cf)
 // https://blogs.msdn.microsoft.com/vcblog/2016/03/30/optimizing-the-layout-of-empty-base-classes-in-vs2015-update-2-3/
 #if PERFETTO_BUILDFLAG(PERFETTO_OS_WIN) && \
-    !PERFETTO_BUILDFLAG(PERFETTO_COMPILER_GCC)
+    PERFETTO_BUILDFLAG(PERFETTO_COMPILER_MSVC)
 #define OPTIONAL_DECLSPEC_EMPTY_BASES __declspec(empty_bases)
 #else
 #define OPTIONAL_DECLSPEC_EMPTY_BASES
diff --git a/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/scoped_file.h b/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/scoped_file.h
index 24c8970..3c14f03 100644
--- a/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/scoped_file.h
+++ b/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/scoped_file.h
@@ -23,7 +23,7 @@
 #include <stdio.h>
 
 #if PERFETTO_BUILDFLAG(PERFETTO_OS_WIN) && \
-    !PERFETTO_BUILDFLAG(PERFETTO_COMPILER_GCC)
+    PERFETTO_BUILDFLAG(PERFETTO_COMPILER_MSVC)
 #include <corecrt_io.h>
 typedef int mode_t;
 #else
diff --git a/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/tracing/core/slice.h b/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/tracing/core/slice.h
index 4ce66fd..3f3932b 100644
--- a/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/tracing/core/slice.h
+++ b/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/tracing/core/slice.h
@@ -20,6 +20,7 @@
 #include <stddef.h>
 #include <string.h>
 
+#include <cstdint>
 #include <memory>
 #include <string>
 #include <vector>
diff --git a/src/3rdparty/chromium/third_party/perfetto/include/perfetto/tracing/internal/track_event_data_source.h b/src/3rdparty/chromium/third_party/perfetto/include/perfetto/tracing/internal/track_event_data_source.h
index bbae795..5a6538f 100644
--- a/src/3rdparty/chromium/third_party/perfetto/include/perfetto/tracing/internal/track_event_data_source.h
+++ b/src/3rdparty/chromium/third_party/perfetto/include/perfetto/tracing/internal/track_event_data_source.h
@@ -107,7 +107,7 @@ class TrackEventDataSource
   }
 
   static void Flush() {
-    Base::template Trace([](typename Base::TraceContext ctx) { ctx.Flush(); });
+    Base::Trace([](typename Base::TraceContext ctx) { ctx.Flush(); });
   }
 
   // Determine if tracing for the given static category is enabled.
@@ -121,7 +121,7 @@ class TrackEventDataSource
   static bool IsDynamicCategoryEnabled(
       const DynamicCategory& dynamic_category) {
     bool enabled = false;
-    Base::template Trace([&](typename Base::TraceContext ctx) {
+    Base::Trace([&](typename Base::TraceContext ctx) {
       enabled = IsDynamicCategoryEnabled(&ctx, dynamic_category);
     });
     return enabled;
@@ -428,7 +428,7 @@ class TrackEventDataSource
                                  const protos::gen::TrackDescriptor& desc) {
     PERFETTO_DCHECK(track.uuid == desc.uuid());
     TrackRegistry::Get()->UpdateTrack(track, desc.SerializeAsString());
-    Base::template Trace([&](typename Base::TraceContext ctx) {
+    Base::Trace([&](typename Base::TraceContext ctx) {
       TrackEventInternal::WriteTrackDescriptor(
           track, ctx.tls_inst_->trace_writer.get());
     });
@@ -545,7 +545,7 @@ class TrackEventDataSource
   static void TraceWithInstances(uint32_t instances,
                                  Lambda lambda) PERFETTO_ALWAYS_INLINE {
     if (CategoryIndex == TrackEventCategoryRegistry::kDynamicCategoryIndex) {
-      Base::template TraceWithInstances(instances, std::move(lambda));
+      Base::TraceWithInstances(instances, std::move(lambda));
     } else {
       Base::template TraceWithInstances<
           CategoryTracePointTraits<CategoryIndex>>(instances,
@@ -560,7 +560,7 @@ class TrackEventDataSource
       const TrackType& track,
       std::function<void(protos::pbzero::TrackDescriptor*)> callback) {
     TrackRegistry::Get()->UpdateTrack(track, std::move(callback));
-    Base::template Trace([&](typename Base::TraceContext ctx) {
+    Base::Trace([&](typename Base::TraceContext ctx) {
       TrackEventInternal::WriteTrackDescriptor(
           track, ctx.tls_inst_->trace_writer.get());
     });
diff --git a/src/3rdparty/chromium/third_party/perfetto/include/perfetto/tracing/tracing_backend.h b/src/3rdparty/chromium/third_party/perfetto/include/perfetto/tracing/tracing_backend.h
index 67b0d2e..3c2462d 100644
--- a/src/3rdparty/chromium/third_party/perfetto/include/perfetto/tracing/tracing_backend.h
+++ b/src/3rdparty/chromium/third_party/perfetto/include/perfetto/tracing/tracing_backend.h
@@ -17,6 +17,7 @@
 #ifndef INCLUDE_PERFETTO_TRACING_TRACING_BACKEND_H_
 #define INCLUDE_PERFETTO_TRACING_TRACING_BACKEND_H_
 
+#include <cstdint>
 #include <memory>
 #include <string>
 
diff --git a/src/3rdparty/chromium/third_party/perfetto/src/base/file_utils.cc b/src/3rdparty/chromium/third_party/perfetto/src/base/file_utils.cc
index b8020ad..fd544e5 100644
--- a/src/3rdparty/chromium/third_party/perfetto/src/base/file_utils.cc
+++ b/src/3rdparty/chromium/third_party/perfetto/src/base/file_utils.cc
@@ -22,7 +22,7 @@
 #include "perfetto/ext/base/scoped_file.h"
 #include "perfetto/ext/base/utils.h"
 
-#if !PERFETTO_BUILDFLAG(PERFETTO_OS_WIN) || \
+#if !PERFETTO_BUILDFLAG(PERFETTO_COMPILER_MSVC) || \
     PERFETTO_BUILDFLAG(PERFETTO_COMPILER_GCC)
 #include <unistd.h>
 #else
diff --git a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/gzip/gzip_utils.h b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/gzip/gzip_utils.h
index 624363f..6d717fb 100644
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/gzip/gzip_utils.h
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/gzip/gzip_utils.h
@@ -17,6 +17,7 @@
 #ifndef SRC_TRACE_PROCESSOR_IMPORTERS_GZIP_GZIP_UTILS_H_
 #define SRC_TRACE_PROCESSOR_IMPORTERS_GZIP_GZIP_UTILS_H_
 
+#include <cstdint>
 #include <memory>
 
 struct z_stream_s;
