diff --git a/src/3rdparty/chromium/third_party/vulkan_memory_allocator/BUILD.gn b/src/3rdparty/chromium/third_party/vulkan_memory_allocator/BUILD.gn
index 8a46926..b0ff4ff 100644
--- a/src/3rdparty/chromium/third_party/vulkan_memory_allocator/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/vulkan_memory_allocator/BUILD.gn
@@ -6,7 +6,7 @@
 
 config("vulkan_memory_allocator_config") {
   include_dirs = [ "src" ]
-  if (is_clang) {
+  if (is_clang || !is_msvc) {
     cflags_cc = [
       "-Wno-implicit-fallthrough",
       "-Wno-nullability-completeness",
@@ -14,7 +14,7 @@ config("vulkan_memory_allocator_config") {
       "-Wno-unused-variable",
     ]
   }
-  if (is_win && !is_clang) {
+  if (is_msvc && !is_clang) {
     cflags_cc = [
       "/wd4189",  # local variable is initialized but not referenced
     ]
