--- a/docs/Makefile.am
+++ b/docs/Makefile.am
@@ -30,7 +30,7 @@
 
 # default is what is needed for interactive gnuplot
 gih_DATA = gnuplot.gih
-gihdir = @GIHDIR@
+gihdir = @datadir@/@PACKAGE@/@VERSION_MAJOR@
 
 noinst_PROGRAMS = checkdoc doc2ipf doc2tex doc2gih doc2hlp \
                   doc2rtf alldoc2gih doc2wxhtml doc2web
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -2,13 +2,13 @@
 AUTOMAKE_OPTIONS = foreign subdir-objects
 
 # in the spirit of automake ...
-pkglibexecdir = $(libexecdir)/@PACKAGE@/@VERSION_MAJOR@
+pkglibexecdir = $(bindir)
 
 # Where to send email about bugs and comments (locally)
 EMAIL = gnuplot-bugs@lists.sourceforge.net
 
 # Default help file location
-HELPFILE=@GIHDIR@/gnuplot.gih
+HELPFILE=$(prefix)/sharex/@PACKAGE@/@VERSION_MAJOR@/gnuplot.gih
 
 # Generic default driver location
 GNUPLOT_DRIVER_DIR=$(pkglibexecdir)
